<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.8.27">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Table of calculations and indices – Biodiverse Calculations and Indices</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./index.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js" type="module"></script>
<script src="site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="site_libs/quarto-html/axe/axe-check.js" type="module"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-ed96de9b727972fe78a7b5d16c58bf87.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-fbb27ab3f4425a88d9682c2047e7f75c.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>

  <script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN" && texText && texText.data) {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

</head>

<body class="nav-sidebar floating quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./indices.html"><span class="chapter-title">Table of calculations and indices</span></a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Biodiverse Calculations and Indices</a> 
        <div class="sidebar-tools-main">
    <a href="./Biodiverse-Calculations-and-Indices.pdf" title="Download PDF" class="quarto-navigation-tool px-1" aria-label="Download PDF"><i class="bi bi-file-pdf"></i></a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link"><span class="chapter-title">Overview</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./indices.html" class="sidebar-item-text sidebar-link active"><span class="chapter-title">Table of calculations and indices</span></a>
  </div>
</li>
    </ul>
    </div>
    <nav id="TOC" role="doc-toc" class="toc-active" data-toc-expanded="99">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#element-properties" id="toc-element-properties" class="nav-link active" data-scroll-target="#element-properties">Element Properties</a>
  <ul class="collapse">
  <li><a href="#group-property-gi-statistics" id="toc-group-property-gi-statistics" class="nav-link" data-scroll-target="#group-property-gi-statistics">Group property Gi* statistics</a></li>
  <li><a href="#group-property-data" id="toc-group-property-data" class="nav-link" data-scroll-target="#group-property-data">Group property data</a></li>
  <li><a href="#group-property-hashes" id="toc-group-property-hashes" class="nav-link" data-scroll-target="#group-property-hashes">Group property hashes</a></li>
  <li><a href="#group-property-quantiles" id="toc-group-property-quantiles" class="nav-link" data-scroll-target="#group-property-quantiles">Group property quantiles</a></li>
  <li><a href="#group-property-summary-stats" id="toc-group-property-summary-stats" class="nav-link" data-scroll-target="#group-property-summary-stats">Group property summary stats</a></li>
  <li><a href="#label-property-gi-statistics" id="toc-label-property-gi-statistics" class="nav-link" data-scroll-target="#label-property-gi-statistics">Label property Gi* statistics</a></li>
  <li><a href="#label-property-gi-statistics-local-range-weighted" id="toc-label-property-gi-statistics-local-range-weighted" class="nav-link" data-scroll-target="#label-property-gi-statistics-local-range-weighted">Label property Gi* statistics (local range weighted)</a></li>
  <li><a href="#label-property-data" id="toc-label-property-data" class="nav-link" data-scroll-target="#label-property-data">Label property data</a></li>
  <li><a href="#label-property-hashes" id="toc-label-property-hashes" class="nav-link" data-scroll-target="#label-property-hashes">Label property hashes</a></li>
  <li><a href="#label-property-hashes-local-range-weighted" id="toc-label-property-hashes-local-range-weighted" class="nav-link" data-scroll-target="#label-property-hashes-local-range-weighted">Label property hashes (local range weighted)</a></li>
  <li><a href="#label-property-lists" id="toc-label-property-lists" class="nav-link" data-scroll-target="#label-property-lists">Label property lists</a></li>
  <li><a href="#label-property-quantiles" id="toc-label-property-quantiles" class="nav-link" data-scroll-target="#label-property-quantiles">Label property quantiles</a></li>
  <li><a href="#label-property-quantiles-local-range-weighted" id="toc-label-property-quantiles-local-range-weighted" class="nav-link" data-scroll-target="#label-property-quantiles-local-range-weighted">Label property quantiles (local range weighted)</a></li>
  <li><a href="#label-property-summary-stats" id="toc-label-property-summary-stats" class="nav-link" data-scroll-target="#label-property-summary-stats">Label property summary stats</a></li>
  <li><a href="#label-property-summary-stats-local-range-weighted" id="toc-label-property-summary-stats-local-range-weighted" class="nav-link" data-scroll-target="#label-property-summary-stats-local-range-weighted">Label property summary stats (local range weighted)</a></li>
  </ul></li>
  <li><a href="#endemism" id="toc-endemism" class="nav-link" data-scroll-target="#endemism">Endemism</a>
  <ul class="collapse">
  <li><a href="#absolute-endemism" id="toc-absolute-endemism" class="nav-link" data-scroll-target="#absolute-endemism">Absolute endemism</a></li>
  <li><a href="#absolute-endemism-lists" id="toc-absolute-endemism-lists" class="nav-link" data-scroll-target="#absolute-endemism-lists">Absolute endemism lists</a></li>
  <li><a href="#endemism-central" id="toc-endemism-central" class="nav-link" data-scroll-target="#endemism-central">Endemism central</a></li>
  <li><a href="#endemism-central-hierarchical-partition" id="toc-endemism-central-hierarchical-partition" class="nav-link" data-scroll-target="#endemism-central-hierarchical-partition">Endemism central hierarchical partition</a></li>
  <li><a href="#endemism-central-lists" id="toc-endemism-central-lists" class="nav-link" data-scroll-target="#endemism-central-lists">Endemism central lists</a></li>
  <li><a href="#endemism-central-normalised" id="toc-endemism-central-normalised" class="nav-link" data-scroll-target="#endemism-central-normalised">Endemism central normalised</a></li>
  <li><a href="#endemism-whole" id="toc-endemism-whole" class="nav-link" data-scroll-target="#endemism-whole">Endemism whole</a></li>
  <li><a href="#endemism-whole-hierarchical-partition" id="toc-endemism-whole-hierarchical-partition" class="nav-link" data-scroll-target="#endemism-whole-hierarchical-partition">Endemism whole hierarchical partition</a></li>
  <li><a href="#endemism-whole-lists" id="toc-endemism-whole-lists" class="nav-link" data-scroll-target="#endemism-whole-lists">Endemism whole lists</a></li>
  <li><a href="#endemism-whole-normalised" id="toc-endemism-whole-normalised" class="nav-link" data-scroll-target="#endemism-whole-normalised">Endemism whole normalised</a></li>
  </ul></li>
  <li><a href="#hierarchical-labels" id="toc-hierarchical-labels" class="nav-link" data-scroll-target="#hierarchical-labels">Hierarchical Labels</a>
  <ul class="collapse">
  <li><a href="#ratios-of-hierarchical-labels" id="toc-ratios-of-hierarchical-labels" class="nav-link" data-scroll-target="#ratios-of-hierarchical-labels">Ratios of hierarchical labels</a></li>
  </ul></li>
  <li><a href="#lists-and-counts" id="toc-lists-and-counts" class="nav-link" data-scroll-target="#lists-and-counts">Lists and Counts</a>
  <ul class="collapse">
  <li><a href="#element-arrays" id="toc-element-arrays" class="nav-link" data-scroll-target="#element-arrays">Element arrays</a></li>
  <li><a href="#element-counts" id="toc-element-counts" class="nav-link" data-scroll-target="#element-counts">Element counts</a></li>
  <li><a href="#element-lists" id="toc-element-lists" class="nav-link" data-scroll-target="#element-lists">Element lists</a></li>
  <li><a href="#label-counts" id="toc-label-counts" class="nav-link" data-scroll-target="#label-counts">Label counts</a></li>
  <li><a href="#label-counts-not-in-sample" id="toc-label-counts-not-in-sample" class="nav-link" data-scroll-target="#label-counts-not-in-sample">Label counts not in sample</a></li>
  <li><a href="#local-range-lists" id="toc-local-range-lists" class="nav-link" data-scroll-target="#local-range-lists">Local range lists</a></li>
  <li><a href="#local-range-summary-statistics" id="toc-local-range-summary-statistics" class="nav-link" data-scroll-target="#local-range-summary-statistics">Local range summary statistics</a></li>
  <li><a href="#non-empty-element-counts" id="toc-non-empty-element-counts" class="nav-link" data-scroll-target="#non-empty-element-counts">Non-empty element counts</a></li>
  <li><a href="#rank-relative-sample-counts-per-label" id="toc-rank-relative-sample-counts-per-label" class="nav-link" data-scroll-target="#rank-relative-sample-counts-per-label">Rank relative sample counts per label</a></li>
  <li><a href="#redundancy" id="toc-redundancy" class="nav-link" data-scroll-target="#redundancy">Redundancy</a></li>
  <li><a href="#richness" id="toc-richness" class="nav-link" data-scroll-target="#richness">Richness</a></li>
  <li><a href="#sample-count-lists" id="toc-sample-count-lists" class="nav-link" data-scroll-target="#sample-count-lists">Sample count lists</a></li>
  <li><a href="#sample-count-quantiles" id="toc-sample-count-quantiles" class="nav-link" data-scroll-target="#sample-count-quantiles">Sample count quantiles</a></li>
  <li><a href="#sample-count-summary-stats" id="toc-sample-count-summary-stats" class="nav-link" data-scroll-target="#sample-count-summary-stats">Sample count summary stats</a></li>
  </ul></li>
  <li><a href="#matrix" id="toc-matrix" class="nav-link" data-scroll-target="#matrix">Matrix</a>
  <ul class="collapse">
  <li><a href="#compare-dissimilarity-matrix-values" id="toc-compare-dissimilarity-matrix-values" class="nav-link" data-scroll-target="#compare-dissimilarity-matrix-values">Compare dissimilarity matrix values</a></li>
  <li><a href="#matrix-statistics" id="toc-matrix-statistics" class="nav-link" data-scroll-target="#matrix-statistics">Matrix statistics</a></li>
  <li><a href="#raos-quadratic-entropy-matrix-weighted" id="toc-raos-quadratic-entropy-matrix-weighted" class="nav-link" data-scroll-target="#raos-quadratic-entropy-matrix-weighted">Rao’s quadratic entropy, matrix weighted</a></li>
  </ul></li>
  <li><a href="#numeric-labels" id="toc-numeric-labels" class="nav-link" data-scroll-target="#numeric-labels">Numeric Labels</a>
  <ul class="collapse">
  <li><a href="#numeric-label-data" id="toc-numeric-label-data" class="nav-link" data-scroll-target="#numeric-label-data">Numeric label data</a></li>
  <li><a href="#numeric-label-dissimilarity" id="toc-numeric-label-dissimilarity" class="nav-link" data-scroll-target="#numeric-label-dissimilarity">Numeric label dissimilarity</a></li>
  <li><a href="#numeric-label-harmonic-and-geometric-means" id="toc-numeric-label-harmonic-and-geometric-means" class="nav-link" data-scroll-target="#numeric-label-harmonic-and-geometric-means">Numeric label harmonic and geometric means</a></li>
  <li><a href="#numeric-label-quantiles" id="toc-numeric-label-quantiles" class="nav-link" data-scroll-target="#numeric-label-quantiles">Numeric label quantiles</a></li>
  <li><a href="#numeric-label-statistics" id="toc-numeric-label-statistics" class="nav-link" data-scroll-target="#numeric-label-statistics">Numeric label statistics</a></li>
  <li><a href="#numeric-labels-gi-statistic" id="toc-numeric-labels-gi-statistic" class="nav-link" data-scroll-target="#numeric-labels-gi-statistic">Numeric labels Gi* statistic</a></li>
  </ul></li>
  <li><a href="#phylocom-indices" id="toc-phylocom-indices" class="nav-link" data-scroll-target="#phylocom-indices">PhyloCom Indices</a>
  <ul class="collapse">
  <li><a href="#nri-and-nti-expected-values" id="toc-nri-and-nti-expected-values" class="nav-link" data-scroll-target="#nri-and-nti-expected-values">NRI and NTI expected values</a></li>
  <li><a href="#nri-and-nti-abundance-weighted" id="toc-nri-and-nti-abundance-weighted" class="nav-link" data-scroll-target="#nri-and-nti-abundance-weighted">NRI and NTI, abundance weighted</a></li>
  <li><a href="#nri-and-nti-local-range-weighted" id="toc-nri-and-nti-local-range-weighted" class="nav-link" data-scroll-target="#nri-and-nti-local-range-weighted">NRI and NTI, local range weighted</a></li>
  <li><a href="#nri-and-nti-unweighted" id="toc-nri-and-nti-unweighted" class="nav-link" data-scroll-target="#nri-and-nti-unweighted">NRI and NTI, unweighted</a></li>
  <li><a href="#net-vpd-expected-values" id="toc-net-vpd-expected-values" class="nav-link" data-scroll-target="#net-vpd-expected-values">Net VPD expected values</a></li>
  <li><a href="#net-variance-of-pair-wise-phylogenetic-distances-unweighted" id="toc-net-variance-of-pair-wise-phylogenetic-distances-unweighted" class="nav-link" data-scroll-target="#net-variance-of-pair-wise-phylogenetic-distances-unweighted">Net variance of pair-wise phylogenetic distances, unweighted</a></li>
  <li><a href="#phylogenetic-and-nearest-taxon-distances-abundance-weighted" id="toc-phylogenetic-and-nearest-taxon-distances-abundance-weighted" class="nav-link" data-scroll-target="#phylogenetic-and-nearest-taxon-distances-abundance-weighted">Phylogenetic and Nearest taxon distances, abundance weighted</a></li>
  <li><a href="#phylogenetic-and-nearest-taxon-distances-local-range-weighted" id="toc-phylogenetic-and-nearest-taxon-distances-local-range-weighted" class="nav-link" data-scroll-target="#phylogenetic-and-nearest-taxon-distances-local-range-weighted">Phylogenetic and Nearest taxon distances, local range weighted</a></li>
  <li><a href="#phylogenetic-and-nearest-taxon-distances-unweighted" id="toc-phylogenetic-and-nearest-taxon-distances-unweighted" class="nav-link" data-scroll-target="#phylogenetic-and-nearest-taxon-distances-unweighted">Phylogenetic and Nearest taxon distances, unweighted</a></li>
  </ul></li>
  <li><a href="#phylogenetic-endemism-indices" id="toc-phylogenetic-endemism-indices" class="nav-link" data-scroll-target="#phylogenetic-endemism-indices">Phylogenetic Endemism Indices</a>
  <ul class="collapse">
  <li><a href="#corrected-weighted-phylogenetic-endemism" id="toc-corrected-weighted-phylogenetic-endemism" class="nav-link" data-scroll-target="#corrected-weighted-phylogenetic-endemism">Corrected weighted phylogenetic endemism</a></li>
  <li><a href="#corrected-weighted-phylogenetic-endemism-central-variant" id="toc-corrected-weighted-phylogenetic-endemism-central-variant" class="nav-link" data-scroll-target="#corrected-weighted-phylogenetic-endemism-central-variant">Corrected weighted phylogenetic endemism, central variant</a></li>
  <li><a href="#corrected-weighted-phylogenetic-rarity" id="toc-corrected-weighted-phylogenetic-rarity" class="nav-link" data-scroll-target="#corrected-weighted-phylogenetic-rarity">Corrected weighted phylogenetic rarity</a></li>
  <li><a href="#pd-endemism" id="toc-pd-endemism" class="nav-link" data-scroll-target="#pd-endemism">PD-Endemism</a></li>
  <li><a href="#pe-clade-contributions" id="toc-pe-clade-contributions" class="nav-link" data-scroll-target="#pe-clade-contributions">PE clade contributions</a></li>
  <li><a href="#pe-clade-loss" id="toc-pe-clade-loss" class="nav-link" data-scroll-target="#pe-clade-loss">PE clade loss</a></li>
  <li><a href="#pe-clade-loss-ancestral-component" id="toc-pe-clade-loss-ancestral-component" class="nav-link" data-scroll-target="#pe-clade-loss-ancestral-component">PE clade loss (ancestral component)</a></li>
  <li><a href="#phylogenetic-endemism" id="toc-phylogenetic-endemism" class="nav-link" data-scroll-target="#phylogenetic-endemism">Phylogenetic Endemism</a></li>
  <li><a href="#phylogenetic-endemism-central" id="toc-phylogenetic-endemism-central" class="nav-link" data-scroll-target="#phylogenetic-endemism-central">Phylogenetic Endemism central</a></li>
  <li><a href="#phylogenetic-endemism-central-lists" id="toc-phylogenetic-endemism-central-lists" class="nav-link" data-scroll-target="#phylogenetic-endemism-central-lists">Phylogenetic Endemism central lists</a></li>
  <li><a href="#phylogenetic-endemism-lists" id="toc-phylogenetic-endemism-lists" class="nav-link" data-scroll-target="#phylogenetic-endemism-lists">Phylogenetic Endemism lists</a></li>
  <li><a href="#phylogenetic-endemism-single" id="toc-phylogenetic-endemism-single" class="nav-link" data-scroll-target="#phylogenetic-endemism-single">Phylogenetic Endemism single</a></li>
  <li><a href="#rwibald" id="toc-rwibald" class="nav-link" data-scroll-target="#rwibald">RWiBaLD</a></li>
  </ul></li>
  <li><a href="#phylogenetic-indices" id="toc-phylogenetic-indices" class="nav-link" data-scroll-target="#phylogenetic-indices">Phylogenetic Indices</a>
  <ul class="collapse">
  <li><a href="#count-labels-on-tree" id="toc-count-labels-on-tree" class="nav-link" data-scroll-target="#count-labels-on-tree">Count labels on tree</a></li>
  <li><a href="#evolutionary-distinctiveness" id="toc-evolutionary-distinctiveness" class="nav-link" data-scroll-target="#evolutionary-distinctiveness">Evolutionary distinctiveness</a></li>
  <li><a href="#evolutionary-distinctiveness-per-site" id="toc-evolutionary-distinctiveness-per-site" class="nav-link" data-scroll-target="#evolutionary-distinctiveness-per-site">Evolutionary distinctiveness per site</a></li>
  <li><a href="#evolutionary-distinctiveness-per-terminal-taxon-per-site" id="toc-evolutionary-distinctiveness-per-terminal-taxon-per-site" class="nav-link" data-scroll-target="#evolutionary-distinctiveness-per-terminal-taxon-per-site">Evolutionary distinctiveness per terminal taxon per site</a></li>
  <li><a href="#labels-not-on-tree" id="toc-labels-not-on-tree" class="nav-link" data-scroll-target="#labels-not-on-tree">Labels not on tree</a></li>
  <li><a href="#labels-on-tree" id="toc-labels-on-tree" class="nav-link" data-scroll-target="#labels-on-tree">Labels on tree</a></li>
  <li><a href="#last-shared-ancestor-properties" id="toc-last-shared-ancestor-properties" class="nav-link" data-scroll-target="#last-shared-ancestor-properties">Last shared ancestor properties</a></li>
  <li><a href="#pd-clade-contributions" id="toc-pd-clade-contributions" class="nav-link" data-scroll-target="#pd-clade-contributions">PD clade contributions</a></li>
  <li><a href="#pd-clade-loss" id="toc-pd-clade-loss" class="nav-link" data-scroll-target="#pd-clade-loss">PD clade loss</a></li>
  <li><a href="#pd-clade-loss-ancestral-component" id="toc-pd-clade-loss-ancestral-component" class="nav-link" data-scroll-target="#pd-clade-loss-ancestral-component">PD clade loss (ancestral component)</a></li>
  <li><a href="#phylogenetic-abundance" id="toc-phylogenetic-abundance" class="nav-link" data-scroll-target="#phylogenetic-abundance">Phylogenetic Abundance</a></li>
  <li><a href="#phylogenetic-diversity" id="toc-phylogenetic-diversity" class="nav-link" data-scroll-target="#phylogenetic-diversity">Phylogenetic Diversity</a></li>
  <li><a href="#phylogenetic-diversity-local" id="toc-phylogenetic-diversity-local" class="nav-link" data-scroll-target="#phylogenetic-diversity-local">Phylogenetic Diversity (local)</a></li>
  <li><a href="#phylogenetic-diversity-node-list" id="toc-phylogenetic-diversity-node-list" class="nav-link" data-scroll-target="#phylogenetic-diversity-node-list">Phylogenetic Diversity node list</a></li>
  <li><a href="#phylogenetic-diversity-terminal-node-count" id="toc-phylogenetic-diversity-terminal-node-count" class="nav-link" data-scroll-target="#phylogenetic-diversity-terminal-node-count">Phylogenetic Diversity terminal node count</a></li>
  <li><a href="#phylogenetic-diversity-terminal-node-list" id="toc-phylogenetic-diversity-terminal-node-list" class="nav-link" data-scroll-target="#phylogenetic-diversity-terminal-node-list">Phylogenetic Diversity terminal node list</a></li>
  </ul></li>
  <li><a href="#phylogenetic-indices-relative" id="toc-phylogenetic-indices-relative" class="nav-link" data-scroll-target="#phylogenetic-indices-relative">Phylogenetic Indices (relative)</a>
  <ul class="collapse">
  <li><a href="#labels-not-on-trimmed-tree" id="toc-labels-not-on-trimmed-tree" class="nav-link" data-scroll-target="#labels-not-on-trimmed-tree">Labels not on trimmed tree</a></li>
  <li><a href="#labels-on-trimmed-tree" id="toc-labels-on-trimmed-tree" class="nav-link" data-scroll-target="#labels-on-trimmed-tree">Labels on trimmed tree</a></li>
  <li><a href="#relative-phylogenetic-diversity-type-1" id="toc-relative-phylogenetic-diversity-type-1" class="nav-link" data-scroll-target="#relative-phylogenetic-diversity-type-1">Relative Phylogenetic Diversity, type 1</a></li>
  <li><a href="#relative-phylogenetic-diversity-type-2" id="toc-relative-phylogenetic-diversity-type-2" class="nav-link" data-scroll-target="#relative-phylogenetic-diversity-type-2">Relative Phylogenetic Diversity, type 2</a></li>
  <li><a href="#relative-phylogenetic-endemism-central" id="toc-relative-phylogenetic-endemism-central" class="nav-link" data-scroll-target="#relative-phylogenetic-endemism-central">Relative Phylogenetic Endemism, central</a></li>
  <li><a href="#relative-phylogenetic-endemism-type-1" id="toc-relative-phylogenetic-endemism-type-1" class="nav-link" data-scroll-target="#relative-phylogenetic-endemism-type-1">Relative Phylogenetic Endemism, type 1</a></li>
  <li><a href="#relative-phylogenetic-endemism-type-2" id="toc-relative-phylogenetic-endemism-type-2" class="nav-link" data-scroll-target="#relative-phylogenetic-endemism-type-2">Relative Phylogenetic Endemism, type 2</a></li>
  </ul></li>
  <li><a href="#phylogenetic-turnover" id="toc-phylogenetic-turnover" class="nav-link" data-scroll-target="#phylogenetic-turnover">Phylogenetic Turnover</a>
  <ul class="collapse">
  <li><a href="#phylo-jaccard" id="toc-phylo-jaccard" class="nav-link" data-scroll-target="#phylo-jaccard">Phylo Jaccard</a></li>
  <li><a href="#phylo-range-weighted-turnover" id="toc-phylo-range-weighted-turnover" class="nav-link" data-scroll-target="#phylo-range-weighted-turnover">Phylo Range weighted Turnover</a></li>
  <li><a href="#phylo-s2" id="toc-phylo-s2" class="nav-link" data-scroll-target="#phylo-s2">Phylo S2</a></li>
  <li><a href="#phylo-sorenson" id="toc-phylo-sorenson" class="nav-link" data-scroll-target="#phylo-sorenson">Phylo Sorenson</a></li>
  <li><a href="#phylogenetic-abc" id="toc-phylogenetic-abc" class="nav-link" data-scroll-target="#phylogenetic-abc">Phylogenetic ABC</a></li>
  </ul></li>
  <li><a href="#rarity" id="toc-rarity" class="nav-link" data-scroll-target="#rarity">Rarity</a>
  <ul class="collapse">
  <li><a href="#rarity-central" id="toc-rarity-central" class="nav-link" data-scroll-target="#rarity-central">Rarity central</a></li>
  <li><a href="#rarity-central-lists" id="toc-rarity-central-lists" class="nav-link" data-scroll-target="#rarity-central-lists">Rarity central lists</a></li>
  <li><a href="#rarity-whole" id="toc-rarity-whole" class="nav-link" data-scroll-target="#rarity-whole">Rarity whole</a></li>
  <li><a href="#rarity-whole-lists" id="toc-rarity-whole-lists" class="nav-link" data-scroll-target="#rarity-whole-lists">Rarity whole lists</a></li>
  </ul></li>
  <li><a href="#richness-estimators" id="toc-richness-estimators" class="nav-link" data-scroll-target="#richness-estimators">Richness estimators</a>
  <ul class="collapse">
  <li><a href="#ace" id="toc-ace" class="nav-link" data-scroll-target="#ace">ACE</a></li>
  <li><a href="#chao1" id="toc-chao1" class="nav-link" data-scroll-target="#chao1">Chao1</a></li>
  <li><a href="#chao2" id="toc-chao2" class="nav-link" data-scroll-target="#chao2">Chao2</a></li>
  <li><a href="#hurlbert-richness-estimation" id="toc-hurlbert-richness-estimation" class="nav-link" data-scroll-target="#hurlbert-richness-estimation">Hurlbert richness estimation</a></li>
  <li><a href="#ice" id="toc-ice" class="nav-link" data-scroll-target="#ice">ICE</a></li>
  </ul></li>
  <li><a href="#taxonomic-dissimilarity-and-comparison" id="toc-taxonomic-dissimilarity-and-comparison" class="nav-link" data-scroll-target="#taxonomic-dissimilarity-and-comparison">Taxonomic Dissimilarity and Comparison</a>
  <ul class="collapse">
  <li><a href="#beta-diversity" id="toc-beta-diversity" class="nav-link" data-scroll-target="#beta-diversity">Beta diversity</a></li>
  <li><a href="#bray-curtis-non-metric" id="toc-bray-curtis-non-metric" class="nav-link" data-scroll-target="#bray-curtis-non-metric">Bray-Curtis non-metric</a></li>
  <li><a href="#bray-curtis-non-metric-group-count-normalised" id="toc-bray-curtis-non-metric-group-count-normalised" class="nav-link" data-scroll-target="#bray-curtis-non-metric-group-count-normalised">Bray-Curtis non-metric, group count normalised</a></li>
  <li><a href="#jaccard" id="toc-jaccard" class="nav-link" data-scroll-target="#jaccard">Jaccard</a></li>
  <li><a href="#kulczynski-2" id="toc-kulczynski-2" class="nav-link" data-scroll-target="#kulczynski-2">Kulczynski 2</a></li>
  <li><a href="#nestedness-resultant" id="toc-nestedness-resultant" class="nav-link" data-scroll-target="#nestedness-resultant">Nestedness-resultant</a></li>
  <li><a href="#range-weighted-sorenson" id="toc-range-weighted-sorenson" class="nav-link" data-scroll-target="#range-weighted-sorenson">Range weighted Sorenson</a></li>
  <li><a href="#raos-quadratic-entropy-taxonomically-weighted" id="toc-raos-quadratic-entropy-taxonomically-weighted" class="nav-link" data-scroll-target="#raos-quadratic-entropy-taxonomically-weighted">Rao’s quadratic entropy, taxonomically weighted</a></li>
  <li><a href="#s2" id="toc-s2" class="nav-link" data-scroll-target="#s2">S2</a></li>
  <li><a href="#simpson-and-shannon" id="toc-simpson-and-shannon" class="nav-link" data-scroll-target="#simpson-and-shannon">Simpson and Shannon</a></li>
  <li><a href="#sorenson" id="toc-sorenson" class="nav-link" data-scroll-target="#sorenson">Sorenson</a></li>
  </ul></li>
  </ul>
</nav>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar zindex-bottom">
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title"><span class="chapter-title">Table of calculations and indices</span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<p><em>Generated GMT Sat Feb 7 06:47:44 2026, Biodiverse version 5.0.</em></p>
<ul>
<li><a href="#element-properties">Element Properties</a>
<ul>
<li><a href="#group-property-gi-statistics">Group property Gi* statistics</a></li>
<li><a href="#group-property-data">Group property data</a></li>
<li><a href="#group-property-hashes">Group property hashes</a></li>
<li><a href="#group-property-quantiles">Group property quantiles</a></li>
<li><a href="#group-property-summary-stats">Group property summary stats</a></li>
<li><a href="#label-property-gi-statistics">Label property Gi* statistics</a></li>
<li><a href="#label-property-gi-statistics-local-range-weighted">Label property Gi* statistics (local range weighted)</a></li>
<li><a href="#label-property-data">Label property data</a></li>
<li><a href="#label-property-hashes">Label property hashes</a></li>
<li><a href="#label-property-hashes-local-range-weighted">Label property hashes (local range weighted)</a></li>
<li><a href="#label-property-lists">Label property lists</a></li>
<li><a href="#label-property-quantiles">Label property quantiles</a></li>
<li><a href="#label-property-quantiles-local-range-weighted">Label property quantiles (local range weighted)</a></li>
<li><a href="#label-property-summary-stats">Label property summary stats</a></li>
<li><a href="#label-property-summary-stats-local-range-weighted">Label property summary stats (local range weighted)</a></li>
</ul></li>
<li><a href="#endemism">Endemism</a>
<ul>
<li><a href="#absolute-endemism">Absolute endemism</a></li>
<li><a href="#absolute-endemism-lists">Absolute endemism lists</a></li>
<li><a href="#endemism-central">Endemism central</a></li>
<li><a href="#endemism-central-hierarchical-partition">Endemism central hierarchical partition</a></li>
<li><a href="#endemism-central-lists">Endemism central lists</a></li>
<li><a href="#endemism-central-normalised">Endemism central normalised</a></li>
<li><a href="#endemism-whole">Endemism whole</a></li>
<li><a href="#endemism-whole-hierarchical-partition">Endemism whole hierarchical partition</a></li>
<li><a href="#endemism-whole-lists">Endemism whole lists</a></li>
<li><a href="#endemism-whole-normalised">Endemism whole normalised</a></li>
</ul></li>
<li><a href="#hierarchical-labels">Hierarchical Labels</a>
<ul>
<li><a href="#ratios-of-hierarchical-labels">Ratios of hierarchical labels</a></li>
</ul></li>
<li><a href="#lists-and-counts">Lists and Counts</a>
<ul>
<li><a href="#element-arrays">Element arrays</a></li>
<li><a href="#element-counts">Element counts</a></li>
<li><a href="#element-lists">Element lists</a></li>
<li><a href="#label-counts">Label counts</a></li>
<li><a href="#label-counts-not-in-sample">Label counts not in sample</a></li>
<li><a href="#local-range-lists">Local range lists</a></li>
<li><a href="#local-range-summary-statistics">Local range summary statistics</a></li>
<li><a href="#non-empty-element-counts">Non-empty element counts</a></li>
<li><a href="#rank-relative-sample-counts-per-label">Rank relative sample counts per label</a></li>
<li><a href="#redundancy">Redundancy</a></li>
<li><a href="#richness">Richness</a></li>
<li><a href="#sample-count-lists">Sample count lists</a></li>
<li><a href="#sample-count-quantiles">Sample count quantiles</a></li>
<li><a href="#sample-count-summary-stats">Sample count summary stats</a></li>
</ul></li>
<li><a href="#matrix">Matrix</a>
<ul>
<li><a href="#compare-dissimilarity-matrix-values">Compare dissimilarity matrix values</a></li>
<li><a href="#matrix-statistics">Matrix statistics</a></li>
<li><a href="#raos-quadratic-entropy-matrix-weighted">Rao’s quadratic entropy, matrix weighted</a></li>
</ul></li>
<li><a href="#numeric-labels">Numeric Labels</a>
<ul>
<li><a href="#numeric-label-data">Numeric label data</a></li>
<li><a href="#numeric-label-dissimilarity">Numeric label dissimilarity</a></li>
<li><a href="#numeric-label-harmonic-and-geometric-means">Numeric label harmonic and geometric means</a></li>
<li><a href="#numeric-label-quantiles">Numeric label quantiles</a></li>
<li><a href="#numeric-label-statistics">Numeric label statistics</a></li>
<li><a href="#numeric-labels-gi-statistic">Numeric labels Gi* statistic</a></li>
</ul></li>
<li><a href="#phylocom-indices">PhyloCom Indices</a>
<ul>
<li><a href="#nri-and-nti-expected-values">NRI and NTI expected values</a></li>
<li><a href="#nri-and-nti-abundance-weighted">NRI and NTI, abundance weighted</a></li>
<li><a href="#nri-and-nti-local-range-weighted">NRI and NTI, local range weighted</a></li>
<li><a href="#nri-and-nti-unweighted">NRI and NTI, unweighted</a></li>
<li><a href="#net-vpd-expected-values">Net VPD expected values</a></li>
<li><a href="#net-variance-of-pair-wise-phylogenetic-distances-unweighted">Net variance of pair-wise phylogenetic distances, unweighted</a></li>
<li><a href="#phylogenetic-and-nearest-taxon-distances-abundance-weighted">Phylogenetic and Nearest taxon distances, abundance weighted</a></li>
<li><a href="#phylogenetic-and-nearest-taxon-distances-local-range-weighted">Phylogenetic and Nearest taxon distances, local range weighted</a></li>
<li><a href="#phylogenetic-and-nearest-taxon-distances-unweighted">Phylogenetic and Nearest taxon distances, unweighted</a></li>
</ul></li>
<li><a href="#phylogenetic-endemism-indices">Phylogenetic Endemism Indices</a>
<ul>
<li><a href="#corrected-weighted-phylogenetic-endemism">Corrected weighted phylogenetic endemism</a></li>
<li><a href="#corrected-weighted-phylogenetic-endemism-central-variant">Corrected weighted phylogenetic endemism, central variant</a></li>
<li><a href="#corrected-weighted-phylogenetic-rarity">Corrected weighted phylogenetic rarity</a></li>
<li><a href="#pd-endemism">PD-Endemism</a></li>
<li><a href="#pe-clade-contributions">PE clade contributions</a></li>
<li><a href="#pe-clade-loss">PE clade loss</a></li>
<li><a href="#pe-clade-loss-ancestral-component">PE clade loss (ancestral component)</a></li>
<li><a href="#phylogenetic-endemism">Phylogenetic Endemism</a></li>
<li><a href="#phylogenetic-endemism-central">Phylogenetic Endemism central</a></li>
<li><a href="#phylogenetic-endemism-central-lists">Phylogenetic Endemism central lists</a></li>
<li><a href="#phylogenetic-endemism-lists">Phylogenetic Endemism lists</a></li>
<li><a href="#phylogenetic-endemism-single">Phylogenetic Endemism single</a></li>
<li><a href="#rwibald">RWiBaLD</a></li>
</ul></li>
<li><a href="#phylogenetic-indices">Phylogenetic Indices</a>
<ul>
<li><a href="#count-labels-on-tree">Count labels on tree</a></li>
<li><a href="#evolutionary-distinctiveness">Evolutionary distinctiveness</a></li>
<li><a href="#evolutionary-distinctiveness-per-site">Evolutionary distinctiveness per site</a></li>
<li><a href="#evolutionary-distinctiveness-per-terminal-taxon-per-site">Evolutionary distinctiveness per terminal taxon per site</a></li>
<li><a href="#labels-not-on-tree">Labels not on tree</a></li>
<li><a href="#labels-on-tree">Labels on tree</a></li>
<li><a href="#last-shared-ancestor-properties">Last shared ancestor properties</a></li>
<li><a href="#pd-clade-contributions">PD clade contributions</a></li>
<li><a href="#pd-clade-loss">PD clade loss</a></li>
<li><a href="#pd-clade-loss-ancestral-component">PD clade loss (ancestral component)</a></li>
<li><a href="#phylogenetic-abundance">Phylogenetic Abundance</a></li>
<li><a href="#phylogenetic-diversity">Phylogenetic Diversity</a></li>
<li><a href="#phylogenetic-diversity-local">Phylogenetic Diversity (local)</a></li>
<li><a href="#phylogenetic-diversity-node-list">Phylogenetic Diversity node list</a></li>
<li><a href="#phylogenetic-diversity-terminal-node-count">Phylogenetic Diversity terminal node count</a></li>
<li><a href="#phylogenetic-diversity-terminal-node-list">Phylogenetic Diversity terminal node list</a></li>
</ul></li>
<li><a href="#phylogenetic-indices-relative">Phylogenetic Indices (relative)</a>
<ul>
<li><a href="#labels-not-on-trimmed-tree">Labels not on trimmed tree</a></li>
<li><a href="#labels-on-trimmed-tree">Labels on trimmed tree</a></li>
<li><a href="#relative-phylogenetic-diversity-type-1">Relative Phylogenetic Diversity, type 1</a></li>
<li><a href="#relative-phylogenetic-diversity-type-2">Relative Phylogenetic Diversity, type 2</a></li>
<li><a href="#relative-phylogenetic-endemism-central">Relative Phylogenetic Endemism, central</a></li>
<li><a href="#relative-phylogenetic-endemism-type-1">Relative Phylogenetic Endemism, type 1</a></li>
<li><a href="#relative-phylogenetic-endemism-type-2">Relative Phylogenetic Endemism, type 2</a></li>
</ul></li>
<li><a href="#phylogenetic-turnover">Phylogenetic Turnover</a>
<ul>
<li><a href="#phylo-jaccard">Phylo Jaccard</a></li>
<li><a href="#phylo-range-weighted-turnover">Phylo Range weighted Turnover</a></li>
<li><a href="#phylo-s2">Phylo S2</a></li>
<li><a href="#phylo-sorenson">Phylo Sorenson</a></li>
<li><a href="#phylogenetic-abc">Phylogenetic ABC</a></li>
</ul></li>
<li><a href="#rarity">Rarity</a>
<ul>
<li><a href="#rarity-central">Rarity central</a></li>
<li><a href="#rarity-central-lists">Rarity central lists</a></li>
<li><a href="#rarity-whole">Rarity whole</a></li>
<li><a href="#rarity-whole-lists">Rarity whole lists</a></li>
</ul></li>
<li><a href="#richness-estimators">Richness estimators</a>
<ul>
<li><a href="#ace">ACE</a></li>
<li><a href="#chao1">Chao1</a></li>
<li><a href="#chao2">Chao2</a></li>
<li><a href="#hurlbert-richness-estimation">Hurlbert richness estimation</a></li>
<li><a href="#ice">ICE</a></li>
</ul></li>
<li><a href="#taxonomic-dissimilarity-and-comparison">Taxonomic Dissimilarity and Comparison</a>
<ul>
<li><a href="#beta-diversity">Beta diversity</a></li>
<li><a href="#bray-curtis-non-metric">Bray-Curtis non-metric</a></li>
<li><a href="#bray-curtis-non-metric-group-count-normalised">Bray-Curtis non-metric, group count normalised</a></li>
<li><a href="#jaccard">Jaccard</a></li>
<li><a href="#kulczynski-2">Kulczynski 2</a></li>
<li><a href="#nestedness-resultant">Nestedness-resultant</a></li>
<li><a href="#range-weighted-sorenson">Range weighted Sorenson</a></li>
<li><a href="#raos-quadratic-entropy-taxonomically-weighted">Rao’s quadratic entropy, taxonomically weighted</a></li>
<li><a href="#s2">S2</a></li>
<li><a href="#simpson-and-shannon">Simpson and Shannon</a></li>
<li><a href="#sorenson">Sorenson</a></li>
</ul></li>
</ul>
<section id="element-properties" class="level2">
<h2 class="anchored" data-anchor-id="element-properties">Element Properties</h2>
<section id="group-property-gi-statistics" class="level3">
<h3 class="anchored" data-anchor-id="group-property-gi-statistics">Group property Gi* statistics</h3>
<p><strong>Description:</strong> List of Getis-Ord Gi* statistics for each group property across both neighbour sets</p>
<p><strong>Subroutine:</strong> calc_gpprop_gistar</p>
<p><strong>Reference:</strong> <a href="https://doi.org/10.1111/j.1538-4632.1992.tb00261.x">Getis and Ord (1992) Geographical Analysis</a></p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>GPPROP_GISTAR_LIST</td>
<td>List of Gi* scores</td>
<td>1</td>
</tr>
</tbody>
</table>
</section>
<section id="group-property-data" class="level3">
<h3 class="anchored" data-anchor-id="group-property-data">Group property data</h3>
<p><strong>Description:</strong> Lists of the groups and their property values used in the group properties calculations. Returns one list for each property, so if your data have properties named ‘GPROP1’ and ‘GPROP2’ then it will return two lists named ‘GPPROP_STATS_GPROP1_DATA’ and ‘GPPROP_STATS_GPROP2_DATA’, respectively.</p>
<p><strong>Subroutine:</strong> calc_gpprop_lists</p>
<p><strong>Indices:</strong></p>
<ul>
<li>Data set dependent</li>
</ul>
</section>
<section id="group-property-hashes" class="level3">
<h3 class="anchored" data-anchor-id="group-property-hashes">Group property hashes</h3>
<p><strong>Description:</strong> Hashes of the groups and their property values used in the group properties calculations. Hash keys are the property values, hash values are the property value frequencies. Returns one list for each property, so if your data have properties named ‘GPROP1’ and ‘GPROP2’ then it will return two lists named ‘GPPROP_STATS_GPROP1_HASH’ and ‘GPPROP_STATS_GPROP2_HASH’, respectively.</p>
<p><strong>Subroutine:</strong> calc_gpprop_hashes</p>
<p><strong>Indices:</strong></p>
<ul>
<li>Data set dependent</li>
</ul>
</section>
<section id="group-property-quantiles" class="level3">
<h3 class="anchored" data-anchor-id="group-property-quantiles">Group property quantiles</h3>
<p><strong>Description:</strong> Quantiles for each group property across both neighbour sets</p>
<p><strong>Subroutine:</strong> calc_gpprop_quantiles</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 33%">
<col style="width: 33%">
<col style="width: 33%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>GPPROP_QUANTILE_LIST</td>
<td>List of quantiles for the label properties (05 10 20 30 40 50 60 70 80 90 95)</td>
<td>1</td>
</tr>
</tbody>
</table>
</section>
<section id="group-property-summary-stats" class="level3">
<h3 class="anchored" data-anchor-id="group-property-summary-stats">Group property summary stats</h3>
<p><strong>Description:</strong> List of summary statistics for each group property across both neighbour sets</p>
<p><strong>Subroutine:</strong> calc_gpprop_stats</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 33%">
<col style="width: 33%">
<col style="width: 33%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>GPPROP_STATS_LIST</td>
<td>List of summary statistics (count mean min max median sum sd iqr)</td>
<td>1</td>
</tr>
</tbody>
</table>
</section>
<section id="label-property-gi-statistics" class="level3">
<h3 class="anchored" data-anchor-id="label-property-gi-statistics">Label property Gi* statistics</h3>
<p><strong>Description:</strong> List of Getis-Ord Gi* statistic for each label property across both neighbour sets</p>
<p><strong>Subroutine:</strong> calc_lbprop_gistar</p>
<p><strong>Reference:</strong> <a href="https://doi.org/10.1111/j.1538-4632.1992.tb00261.x">Getis and Ord (1992) Geographical Analysis</a></p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>LBPROP_GISTAR_LIST</td>
<td>List of Gi* scores</td>
<td>1</td>
</tr>
</tbody>
</table>
</section>
<section id="label-property-gi-statistics-local-range-weighted" class="level3">
<h3 class="anchored" data-anchor-id="label-property-gi-statistics-local-range-weighted">Label property Gi* statistics (local range weighted)</h3>
<p><strong>Description:</strong> List of Getis-Ord Gi* statistic values for each label property across both neighbour sets (local range weighted)</p>
<p><strong>Subroutine:</strong> calc_lbprop_gistar_abc2</p>
<p><strong>Reference:</strong> <a href="https://doi.org/10.1111/j.1538-4632.1992.tb00261.x">Getis and Ord (1992) Geographical Analysis</a></p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>LBPROP_GISTAR_LIST_ABC2</td>
<td>List of Gi* scores</td>
<td>1</td>
</tr>
</tbody>
</table>
</section>
<section id="label-property-data" class="level3">
<h3 class="anchored" data-anchor-id="label-property-data">Label property data</h3>
<p><strong>Description:</strong> Lists of the labels and their property values used in the label properties calculations. Returns one list for each property, so if your data have properties named ‘PROP1’ and ‘PROP2’ then it will return two lists named ‘LBPROP_STATS_PROP1_DATA’ and ‘LBPROP_STATS_PROP1_DATA’, respectively.</p>
<p><strong>Subroutine:</strong> calc_lbprop_data</p>
<p><strong>Indices:</strong></p>
<ul>
<li>Data set dependent</li>
</ul>
</section>
<section id="label-property-hashes" class="level3">
<h3 class="anchored" data-anchor-id="label-property-hashes">Label property hashes</h3>
<p><strong>Description:</strong> Hashes of the labels and their property values used in the label properties calculations. Hash keys are the property values, hash values are the property value frequencies. Returns one hash for each property, so if your data have properties named ‘PROP1’ and ‘PROP2’ then it will return two lists named ‘LBPROP_STATS_PROP1_HASH’ and ‘LBPROP_STATS_PROP2_HASH’, respectively.</p>
<p><strong>Subroutine:</strong> calc_lbprop_hashes</p>
<p><strong>Indices:</strong></p>
<ul>
<li>Data set dependent</li>
</ul>
</section>
<section id="label-property-hashes-local-range-weighted" class="level3">
<h3 class="anchored" data-anchor-id="label-property-hashes-local-range-weighted">Label property hashes (local range weighted)</h3>
<p><strong>Description:</strong> Hashes of the labels and their property values used in the local range weighted label properties calculations. Hash keys are the property values, hash values are the property value frequencies. Returns one hash for each property, so if your data have properties named ‘PROP1’ and ‘PROP2’ then it will return two lists named ‘LBPROP_STATS_PROP1_HASH2’ and ‘LBPROP_STATS_PROP2_HASH2’, respectively.</p>
<p><strong>Subroutine:</strong> calc_lbprop_hashes_abc2</p>
<p><strong>Indices:</strong></p>
<ul>
<li>Data set dependent</li>
</ul>
</section>
<section id="label-property-lists" class="level3">
<h3 class="anchored" data-anchor-id="label-property-lists">Label property lists</h3>
<p><strong>Description:</strong> Lists of the labels and their property values within the neighbour sets. Returns one list for each property, so if your data have properties named ‘PROP1’ and ‘PROP2’ then it will return two lists named ‘LBPROP_LIST_PROP1’ and ‘LBPROP_LIST_PROP2’, respectively.</p>
<p><strong>Subroutine:</strong> calc_lbprop_lists</p>
<p><strong>Indices:</strong></p>
<ul>
<li>Data set dependent</li>
</ul>
</section>
<section id="label-property-quantiles" class="level3">
<h3 class="anchored" data-anchor-id="label-property-quantiles">Label property quantiles</h3>
<p><strong>Description:</strong> List of quantiles for each label property across both neighbour sets</p>
<p><strong>Subroutine:</strong> calc_lbprop_quantiles</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 33%">
<col style="width: 33%">
<col style="width: 33%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>LBPROP_QUANTILES</td>
<td>List of quantiles for the label properties: (05 10 20 30 40 50 60 70 80 90 95)</td>
<td>1</td>
</tr>
</tbody>
</table>
</section>
<section id="label-property-quantiles-local-range-weighted" class="level3">
<h3 class="anchored" data-anchor-id="label-property-quantiles-local-range-weighted">Label property quantiles (local range weighted)</h3>
<p><strong>Description:</strong> List of quantiles for each label property across both neighbour sets (local range weighted)</p>
<p><strong>Subroutine:</strong> calc_lbprop_quantiles_abc2</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 33%">
<col style="width: 33%">
<col style="width: 33%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>LBPROP_QUANTILES_ABC2</td>
<td>List of quantiles for the label properties: (05 10 20 30 40 50 60 70 80 90 95)</td>
<td>1</td>
</tr>
</tbody>
</table>
</section>
<section id="label-property-summary-stats" class="level3">
<h3 class="anchored" data-anchor-id="label-property-summary-stats">Label property summary stats</h3>
<p><strong>Description:</strong> List of summary statistics for each label property across both neighbour sets</p>
<p><strong>Subroutine:</strong> calc_lbprop_stats</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 33%">
<col style="width: 33%">
<col style="width: 33%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>LBPROP_STATS</td>
<td>List of summary statistics (count mean min max median sum skewness kurtosis sd iqr)</td>
<td>1</td>
</tr>
</tbody>
</table>
</section>
<section id="label-property-summary-stats-local-range-weighted" class="level3">
<h3 class="anchored" data-anchor-id="label-property-summary-stats-local-range-weighted">Label property summary stats (local range weighted)</h3>
<p><strong>Description:</strong> List of summary statistics for each label property across both neighbour sets, weighted by local ranges</p>
<p><strong>Subroutine:</strong> calc_lbprop_stats_abc2</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 33%">
<col style="width: 33%">
<col style="width: 33%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>LBPROP_STATS_ABC2</td>
<td>List of summary statistics (count mean min max median sum skewness kurtosis sd iqr)</td>
<td>1</td>
</tr>
</tbody>
</table>
</section>
</section>
<section id="endemism" class="level2">
<h2 class="anchored" data-anchor-id="endemism">Endemism</h2>
<section id="absolute-endemism" class="level3">
<h3 class="anchored" data-anchor-id="absolute-endemism">Absolute endemism</h3>
<p><strong>Description:</strong> Absolute endemism scores.</p>
<p><strong>Subroutine:</strong> calc_endemism_absolute</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Grouping metric?</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>END_ABS1</td>
<td>Count of labels entirely endemic to neighbour set 1</td>
<td></td>
<td>1</td>
</tr>
<tr class="even">
<td>END_ABS1_P</td>
<td>Proportion of labels entirely endemic to neighbour set 1</td>
<td></td>
<td>1</td>
</tr>
<tr class="odd">
<td>END_ABS2</td>
<td>Count of labels entirely endemic to neighbour set 2</td>
<td></td>
<td>1</td>
</tr>
<tr class="even">
<td>END_ABS2_P</td>
<td>Proportion of labels entirely endemic to neighbour set 2</td>
<td></td>
<td>1</td>
</tr>
<tr class="odd">
<td>END_ABS_ALL</td>
<td>Count of labels entirely endemic to neighbour sets 1 and 2 combined</td>
<td>region grower</td>
<td>1</td>
</tr>
<tr class="even">
<td>END_ABS_ALL_P</td>
<td>Proportion of labels entirely endemic to neighbour sets 1 and 2 combined</td>
<td></td>
<td>1</td>
</tr>
</tbody>
</table>
</section>
<section id="absolute-endemism-lists" class="level3">
<h3 class="anchored" data-anchor-id="absolute-endemism-lists">Absolute endemism lists</h3>
<p><strong>Description:</strong> Lists underlying the absolute endemism scores.</p>
<p><strong>Subroutine:</strong> calc_endemism_absolute_lists</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 33%">
<col style="width: 33%">
<col style="width: 33%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>END_ABS1_LIST</td>
<td>List of labels entirely endemic to neighbour set 1</td>
<td>1</td>
</tr>
<tr class="even">
<td>END_ABS2_LIST</td>
<td>List of labels entirely endemic to neighbour set 1</td>
<td>1</td>
</tr>
<tr class="odd">
<td>END_ABS_ALL_LIST</td>
<td>List of labels entirely endemic to neighbour sets 1 and 2 combined</td>
<td>1</td>
</tr>
</tbody>
</table>
</section>
<section id="endemism-central" class="level3">
<h3 class="anchored" data-anchor-id="endemism-central">Endemism central</h3>
<p><strong>Description:</strong> Calculate endemism for labels only in neighbour set 1, but with local ranges calculated using both neighbour sets</p>
<p><strong>Subroutine:</strong> calc_endemism_central</p>
<p><strong>Reference:</strong> <a href="https://doi.org/10.1046/j.1365-2699.2001.00524.x">Crisp et al.&nbsp;(2001) J Biogeog</a>; <a href="http://www3.interscience.wiley.com/journal/118882020/abstract">Laffan and Crisp (2003) J Biogeog</a></p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
<th><em>Formula</em></th>
<th><em>Reference</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>ENDC_CWE</td>
<td>Corrected weighted endemism</td>
<td>1</td>
<td><span class="math inline">\(= \frac{ENDC\_WE}{ENDC\_RICHNESS}\)</span></td>
<td></td>
</tr>
<tr class="even">
<td>ENDC_RICHNESS</td>
<td>Richness used in ENDC_CWE (same as index RICHNESS_SET1)</td>
<td>1</td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>ENDC_SINGLE</td>
<td>Endemism unweighted by the number of neighbours. Counts each label only once, regardless of how many groups in the neighbourhood it is found in. Useful if your data have sampling biases and best applied with a small window.</td>
<td>1</td>
<td><span class="math inline">\(= \sum_{t \in T} \frac {1} {R_t}\)</span> where <span class="math inline">\(t\)</span> is a label (taxon) in the set of labels (taxa) <span class="math inline">\(T\)</span> in neighbour set 1, and <span class="math inline">\(R_t\)</span> is the global range of label <span class="math inline">\(t\)</span> across the data set (the number of groups it is found in, unless the range is specified at import).</td>
<td><a href="https://doi.org/10.1111/j.1365-2699.2006.01647.x">Slatyer et al.&nbsp;(2007) J. Biogeog</a></td>
</tr>
<tr class="even">
<td>ENDC_WE</td>
<td>Weighted endemism</td>
<td>1</td>
<td><span class="math inline">\(= \sum_{t \in T} \frac {r_t} {R_t}\)</span> where <span class="math inline">\(t\)</span> is a label (taxon) in the set of labels (taxa) <span class="math inline">\(T\)</span> in neighbour set 1, <span class="math inline">\(r_t\)</span> is the local range (the number of elements containing label <span class="math inline">\(t\)</span> within neighbour sets 1 &amp; 2, this is also its value in list ABC2_LABELS_ALL), and <span class="math inline">\(R_t\)</span> is the global range of label <span class="math inline">\(t\)</span> across the data set (the number of groups it is found in, unless the range is specified at import).</td>
<td></td>
</tr>
</tbody>
</table>
</section>
<section id="endemism-central-hierarchical-partition" class="level3">
<h3 class="anchored" data-anchor-id="endemism-central-hierarchical-partition">Endemism central hierarchical partition</h3>
<p><strong>Description:</strong> Partition the endemism central results based on the taxonomic hierarchy inferred from the label axes. (Level 0 is the highest).</p>
<p><strong>Subroutine:</strong> calc_endemism_central_hier_part</p>
<p><strong>Reference:</strong> <a href="https://doi.org/10.1111/jbi.12001">Laffan et al.&nbsp;(2013) J Biogeog</a></p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 33%">
<col style="width: 33%">
<col style="width: 33%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>ENDC_HPART_0</td>
<td>List of the proportional contribution of labels to the endemism central calculations, hierarchical level 0</td>
<td>1</td>
</tr>
<tr class="even">
<td>ENDC_HPART_1</td>
<td>List of the proportional contribution of labels to the endemism central calculations, hierarchical level 1</td>
<td>1</td>
</tr>
<tr class="odd">
<td>ENDC_HPART_C_0</td>
<td>List of the proportional count of labels to the endemism central calculations (equivalent to richness per hierarchical grouping), hierarchical level 0</td>
<td>1</td>
</tr>
<tr class="even">
<td>ENDC_HPART_C_1</td>
<td>List of the proportional count of labels to the endemism central calculations (equivalent to richness per hierarchical grouping), hierarchical level 1</td>
<td>1</td>
</tr>
<tr class="odd">
<td>ENDC_HPART_E_0</td>
<td>List of the expected proportional contribution of labels to the endemism central calculations (richness per hierarchical grouping divided by overall richness), hierarchical level 0</td>
<td>1</td>
</tr>
<tr class="even">
<td>ENDC_HPART_E_1</td>
<td>List of the expected proportional contribution of labels to the endemism central calculations (richness per hierarchical grouping divided by overall richness), hierarchical level 1</td>
<td>1</td>
</tr>
<tr class="odd">
<td>ENDC_HPART_OME_0</td>
<td>List of the observed minus expected proportional contribution of labels to the endemism central calculations , hierarchical level 0</td>
<td>1</td>
</tr>
<tr class="even">
<td>ENDC_HPART_OME_1</td>
<td>List of the observed minus expected proportional contribution of labels to the endemism central calculations , hierarchical level 1</td>
<td>1</td>
</tr>
</tbody>
</table>
</section>
<section id="endemism-central-lists" class="level3">
<h3 class="anchored" data-anchor-id="endemism-central-lists">Endemism central lists</h3>
<p><strong>Description:</strong> Lists used in endemism central calculations</p>
<p><strong>Subroutine:</strong> calc_endemism_central_lists</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 33%">
<col style="width: 33%">
<col style="width: 33%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>ENDC_RANGELIST</td>
<td>List of ranges for each label used in the endemism central calculations</td>
<td>1</td>
</tr>
<tr class="even">
<td>ENDC_WTLIST</td>
<td>List of weights for each label used in the endemism central calculations</td>
<td>1</td>
</tr>
</tbody>
</table>
</section>
<section id="endemism-central-normalised" class="level3">
<h3 class="anchored" data-anchor-id="endemism-central-normalised">Endemism central normalised</h3>
<p><strong>Description:</strong> Normalise the WE and CWE scores by the neighbourhood size. (The number of groups used to determine the local ranges).</p>
<p><strong>Subroutine:</strong> calc_endemism_central_normalised</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
<th><em>Formula</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>ENDC_CWE_NORM</td>
<td>Corrected weighted endemism normalised by groups</td>
<td>1</td>
<td><span class="math inline">\(= \frac{ENDC\_CWE}{EL\_COUNT\_ALL}\)</span></td>
</tr>
<tr class="even">
<td>ENDC_WE_NORM</td>
<td>Weighted endemism normalised by groups</td>
<td>1</td>
<td><span class="math inline">\(= \frac{ENDC\_WE}{EL\_COUNT\_ALL}\)</span></td>
</tr>
</tbody>
</table>
</section>
<section id="endemism-whole" class="level3">
<h3 class="anchored" data-anchor-id="endemism-whole">Endemism whole</h3>
<p><strong>Description:</strong> Calculate endemism using all labels found in both neighbour sets</p>
<p><strong>Subroutine:</strong> calc_endemism_whole</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Grouping metric?</em></th>
<th><em>Minimum number of neighbour sets</em></th>
<th><em>Formula</em></th>
<th><em>Reference</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>ENDW_CWE</td>
<td>Corrected weighted endemism</td>
<td></td>
<td>1</td>
<td><span class="math inline">\(= \frac{ENDW\_WE}{ENDW\_RICHNESS}\)</span></td>
<td></td>
</tr>
<tr class="even">
<td>ENDW_RICHNESS</td>
<td>Richness used in ENDW_CWE (same as index RICHNESS_ALL)</td>
<td>region grower</td>
<td>1</td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>ENDW_SINGLE</td>
<td>Endemism unweighted by the number of neighbours. Counts each label only once, regardless of how many groups in the neighbourhood it is found in. Useful if your data have sampling biases and best applied with a small window.</td>
<td>region grower</td>
<td>1</td>
<td><span class="math inline">\(= \sum_{t \in T} \frac {1} {R_t}\)</span> where <span class="math inline">\(t\)</span> is a label (taxon) in the set of labels (taxa) <span class="math inline">\(T\)</span> across neighbour sets 1 &amp; 2, and <span class="math inline">\(R_t\)</span> is the global range of label <span class="math inline">\(t\)</span> across the data set (the number of groups it is found in, unless the range is specified at import).</td>
<td><a href="https://doi.org/10.1111/j.1365-2699.2006.01647.x">Slatyer et al.&nbsp;(2007) J. Biogeog</a></td>
</tr>
<tr class="even">
<td>ENDW_WE</td>
<td>Weighted endemism</td>
<td>region grower</td>
<td>1</td>
<td><span class="math inline">\(= \sum_{t \in T} \frac {r_t} {R_t}\)</span> where <span class="math inline">\(t\)</span> is a label (taxon) in the set of labels (taxa) <span class="math inline">\(T\)</span> across both neighbour sets, <span class="math inline">\(r_t\)</span> is the local range (the number of elements containing label <span class="math inline">\(t\)</span> within neighbour sets 1 &amp; 2, this is also its value in list ABC2_LABELS_ALL), and <span class="math inline">\(R_t\)</span> is the global range of label <span class="math inline">\(t\)</span> across the data set (the number of groups it is found in, unless the range is specified at import).</td>
<td></td>
</tr>
</tbody>
</table>
</section>
<section id="endemism-whole-hierarchical-partition" class="level3">
<h3 class="anchored" data-anchor-id="endemism-whole-hierarchical-partition">Endemism whole hierarchical partition</h3>
<p><strong>Description:</strong> Partition the endemism whole results based on the taxonomic hierarchy inferred from the label axes. (Level 0 is the highest).</p>
<p><strong>Subroutine:</strong> calc_endemism_whole_hier_part</p>
<p><strong>Reference:</strong> <a href="https://doi.org/10.1111/jbi.12001">Laffan et al.&nbsp;(2013) J Biogeog</a></p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 33%">
<col style="width: 33%">
<col style="width: 33%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>ENDW_HPART_0</td>
<td>List of the proportional contribution of labels to the endemism whole calculations, hierarchical level 0</td>
<td>1</td>
</tr>
<tr class="even">
<td>ENDW_HPART_1</td>
<td>List of the proportional contribution of labels to the endemism whole calculations, hierarchical level 1</td>
<td>1</td>
</tr>
<tr class="odd">
<td>ENDW_HPART_C_0</td>
<td>List of the proportional count of labels to the endemism whole calculations (equivalent to richness per hierarchical grouping), hierarchical level 0</td>
<td>1</td>
</tr>
<tr class="even">
<td>ENDW_HPART_C_1</td>
<td>List of the proportional count of labels to the endemism whole calculations (equivalent to richness per hierarchical grouping), hierarchical level 1</td>
<td>1</td>
</tr>
<tr class="odd">
<td>ENDW_HPART_E_0</td>
<td>List of the expected proportional contribution of labels to the endemism whole calculations (richness per hierarchical grouping divided by overall richness), hierarchical level 0</td>
<td>1</td>
</tr>
<tr class="even">
<td>ENDW_HPART_E_1</td>
<td>List of the expected proportional contribution of labels to the endemism whole calculations (richness per hierarchical grouping divided by overall richness), hierarchical level 1</td>
<td>1</td>
</tr>
<tr class="odd">
<td>ENDW_HPART_OME_0</td>
<td>List of the observed minus expected proportional contribution of labels to the endemism whole calculations , hierarchical level 0</td>
<td>1</td>
</tr>
<tr class="even">
<td>ENDW_HPART_OME_1</td>
<td>List of the observed minus expected proportional contribution of labels to the endemism whole calculations , hierarchical level 1</td>
<td>1</td>
</tr>
</tbody>
</table>
</section>
<section id="endemism-whole-lists" class="level3">
<h3 class="anchored" data-anchor-id="endemism-whole-lists">Endemism whole lists</h3>
<p><strong>Description:</strong> Lists used in the endemism whole calculations</p>
<p><strong>Subroutine:</strong> calc_endemism_whole_lists</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 33%">
<col style="width: 33%">
<col style="width: 33%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>ENDW_RANGELIST</td>
<td>List of ranges for each label used in the endemism whole calculations</td>
<td>1</td>
</tr>
<tr class="even">
<td>ENDW_WTLIST</td>
<td>List of weights for each label used in the endemism whole calculations</td>
<td>1</td>
</tr>
</tbody>
</table>
</section>
<section id="endemism-whole-normalised" class="level3">
<h3 class="anchored" data-anchor-id="endemism-whole-normalised">Endemism whole normalised</h3>
<p><strong>Description:</strong> Normalise the WE and CWE scores by the neighbourhood size. (The number of groups used to determine the local ranges).</p>
<p><strong>Subroutine:</strong> calc_endemism_whole_normalised</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
<th><em>Formula</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>ENDW_CWE_NORM</td>
<td>Corrected weighted endemism normalised by groups</td>
<td>1</td>
<td><span class="math inline">\(= \frac{ENDW\_CWE}{EL\_COUNT\_ALL}\)</span></td>
</tr>
<tr class="even">
<td>ENDW_WE_NORM</td>
<td>Weighted endemism normalised by groups</td>
<td>1</td>
<td><span class="math inline">\(= \frac{ENDW\_WE}{EL\_COUNT\_ALL}\)</span></td>
</tr>
</tbody>
</table>
</section>
</section>
<section id="hierarchical-labels" class="level2">
<h2 class="anchored" data-anchor-id="hierarchical-labels">Hierarchical Labels</h2>
<section id="ratios-of-hierarchical-labels" class="level3">
<h3 class="anchored" data-anchor-id="ratios-of-hierarchical-labels">Ratios of hierarchical labels</h3>
<p><strong>Description:</strong> Analyse the diversity of labels using their hierarchical levels. The A, B and C scores are the same as in the Label Counts analysis (calc_label_counts) but calculated for each hierarchical level, e.g.&nbsp;for three axes one could have A0 as the Family level, A1 for the Family:Genus level, and A2 for the Family:Genus:Species level. The number of list elements generated depends on how many axes are used in the labels. Axes are order from zero as the highest level in the hierarchy, so index 0 is the top level of the hierarchy.</p>
<p>Note that this calculation produces lists since version 4.99_002 so one can no longer use the SUMRAT indices for clustering. This can be re-enabled if there is a need.</p>
<p><strong>Subroutine:</strong> calc_hierarchical_label_ratios</p>
<p><strong>Reference:</strong> <a href="https://doi.org/10.1111/j.1467-968X.2008.00209.x">Jones and Laffan (2008) Trans Philol Soc</a></p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 33%">
<col style="width: 33%">
<col style="width: 33%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>HIER_A</td>
<td>A score for each level</td>
<td>2</td>
</tr>
<tr class="even">
<td>HIER_ARAT</td>
<td>Ratio of A scores between adjacent levels</td>
<td>2</td>
</tr>
<tr class="odd">
<td>HIER_ASUM</td>
<td>Sum of shared label sample counts</td>
<td>2</td>
</tr>
<tr class="even">
<td>HIER_ASUMRAT</td>
<td>1 - Ratio of shared label sample counts between adjacent levels</td>
<td>2</td>
</tr>
<tr class="odd">
<td>HIER_B</td>
<td>B score for each level</td>
<td>2</td>
</tr>
<tr class="even">
<td>HIER_BRAT</td>
<td>Ratio of B scores between adjacent levels</td>
<td>2</td>
</tr>
<tr class="odd">
<td>HIER_C</td>
<td>C score for each level</td>
<td>2</td>
</tr>
<tr class="even">
<td>HIER_CRAT</td>
<td>Ratio of C scores between adjacent levels</td>
<td>2</td>
</tr>
</tbody>
</table>
</section>
</section>
<section id="lists-and-counts" class="level2">
<h2 class="anchored" data-anchor-id="lists-and-counts">Lists and Counts</h2>
<section id="element-arrays" class="level3">
<h3 class="anchored" data-anchor-id="element-arrays">Element arrays</h3>
<p><strong>Description:</strong> Arrays of elements used in neighbour sets 1 and 2. These form the basis for all the spatial calculations.</p>
<p><strong>Subroutine:</strong> calc_element_lists_used_as_arrays</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>EL_ARRAY_ALL</td>
<td>Array of elements in both neighbour sets</td>
<td>2</td>
</tr>
<tr class="even">
<td>EL_ARRAY_SET1</td>
<td>Array of elements in neighbour set 1</td>
<td>1</td>
</tr>
<tr class="odd">
<td>EL_ARRAY_SET2</td>
<td>Array of elements in neighbour set 2</td>
<td>2</td>
</tr>
</tbody>
</table>
</section>
<section id="element-counts" class="level3">
<h3 class="anchored" data-anchor-id="element-counts">Element counts</h3>
<p><strong>Description:</strong> Counts of elements used in neighbour sets 1 and 2.</p>
<p><strong>Subroutine:</strong> calc_elements_used</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Grouping metric?</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>EL_COUNT_ALL</td>
<td>Count of elements in both neighbour sets</td>
<td>region grower</td>
<td>1</td>
</tr>
<tr class="even">
<td>EL_COUNT_SET1</td>
<td>Count of elements in neighbour set 1</td>
<td></td>
<td>1</td>
</tr>
<tr class="odd">
<td>EL_COUNT_SET2</td>
<td>Count of elements in neighbour set 2</td>
<td></td>
<td>2</td>
</tr>
</tbody>
</table>
</section>
<section id="element-lists" class="level3">
<h3 class="anchored" data-anchor-id="element-lists">Element lists</h3>
<p><strong>Description:</strong> [DEPRECATED] Lists of elements used in neighbour sets 1 and 2. These form the basis for all the spatial calculations. The return types are inconsistent. New code should use calc_element_lists_used_as_arrays</p>
<p><strong>Subroutine:</strong> calc_element_lists_used</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>EL_LIST_ALL</td>
<td>List of elements in both neighour sets</td>
<td>2</td>
</tr>
<tr class="even">
<td>EL_LIST_SET1</td>
<td>List of elements in neighbour set 1</td>
<td>1</td>
</tr>
<tr class="odd">
<td>EL_LIST_SET2</td>
<td>List of elements in neighbour set 2</td>
<td>2</td>
</tr>
</tbody>
</table>
</section>
<section id="label-counts" class="level3">
<h3 class="anchored" data-anchor-id="label-counts">Label counts</h3>
<p><strong>Description:</strong> Counts of labels in neighbour sets 1 and 2. These form the basis for the Taxonomic Dissimilarity and Comparison indices.</p>
<p><strong>Subroutine:</strong> calc_abc_counts</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Grouping metric?</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>ABC_A</td>
<td>Count of labels common to both neighbour sets</td>
<td>region grower</td>
<td>2</td>
</tr>
<tr class="even">
<td>ABC_ABC</td>
<td>Total label count across both neighbour sets (same as RICHNESS_ALL)</td>
<td>region grower</td>
<td>2</td>
</tr>
<tr class="odd">
<td>ABC_B</td>
<td>Count of labels unique to neighbour set 1</td>
<td></td>
<td>2</td>
</tr>
<tr class="even">
<td>ABC_C</td>
<td>Count of labels unique to neighbour set 2</td>
<td></td>
<td>2</td>
</tr>
</tbody>
</table>
</section>
<section id="label-counts-not-in-sample" class="level3">
<h3 class="anchored" data-anchor-id="label-counts-not-in-sample">Label counts not in sample</h3>
<p><strong>Description:</strong> Count of basedata labels not in either neighbour set (shared absence) Used in some of the dissimilarity metrics.</p>
<p><strong>Subroutine:</strong> calc_d</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>ABC_D</td>
<td>Count of labels not in either neighbour set (D score)</td>
<td>1</td>
</tr>
</tbody>
</table>
</section>
<section id="local-range-lists" class="level3">
<h3 class="anchored" data-anchor-id="local-range-lists">Local range lists</h3>
<p><strong>Description:</strong> Lists of labels with their local ranges as values. The local ranges are the number of elements in which each label is found in each neighour set.</p>
<p><strong>Subroutine:</strong> calc_local_range_lists</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>ABC2_LABELS_ALL</td>
<td>List of labels in both neighbour sets</td>
<td>2</td>
</tr>
<tr class="even">
<td>ABC2_LABELS_SET1</td>
<td>List of labels in neighbour set 1</td>
<td>1</td>
</tr>
<tr class="odd">
<td>ABC2_LABELS_SET2</td>
<td>List of labels in neighbour set 2</td>
<td>2</td>
</tr>
</tbody>
</table>
</section>
<section id="local-range-summary-statistics" class="level3">
<h3 class="anchored" data-anchor-id="local-range-summary-statistics">Local range summary statistics</h3>
<p><strong>Description:</strong> Summary stats of the local ranges within neighour sets.</p>
<p><strong>Subroutine:</strong> calc_local_range_stats</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 33%">
<col style="width: 33%">
<col style="width: 33%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>ABC2_MEAN_ALL</td>
<td>Mean label range in both element sets</td>
<td>1</td>
</tr>
<tr class="even">
<td>ABC2_MEAN_SET1</td>
<td>Mean label range in neighbour set 1</td>
<td>1</td>
</tr>
<tr class="odd">
<td>ABC2_MEAN_SET2</td>
<td>Mean label range in neighbour set 2</td>
<td>2</td>
</tr>
<tr class="even">
<td>ABC2_SD_ALL</td>
<td>Standard deviation of label ranges in both element sets</td>
<td>2</td>
</tr>
<tr class="odd">
<td>ABC2_SD_SET1</td>
<td>Standard deviation of label ranges in neighbour set 1</td>
<td>1</td>
</tr>
<tr class="even">
<td>ABC2_SD_SET2</td>
<td>Standard deviation of label ranges in neighbour set 2</td>
<td>2</td>
</tr>
</tbody>
</table>
</section>
<section id="non-empty-element-counts" class="level3">
<h3 class="anchored" data-anchor-id="non-empty-element-counts">Non-empty element counts</h3>
<p><strong>Description:</strong> Counts of non-empty elements in neighbour sets 1 and 2.</p>
<p><strong>Subroutine:</strong> calc_nonempty_elements_used</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 33%">
<col style="width: 33%">
<col style="width: 33%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>EL_COUNT_NONEMPTY_ALL</td>
<td>Count of non-empty elements in both neighbour sets</td>
<td>1</td>
</tr>
<tr class="even">
<td>EL_COUNT_NONEMPTY_SET1</td>
<td>Count of non-empty elements in neighbour set 1</td>
<td>1</td>
</tr>
<tr class="odd">
<td>EL_COUNT_NONEMPTY_SET2</td>
<td>Count of non-empty elements in neighbour set 2</td>
<td>2</td>
</tr>
</tbody>
</table>
</section>
<section id="rank-relative-sample-counts-per-label" class="level3">
<h3 class="anchored" data-anchor-id="rank-relative-sample-counts-per-label">Rank relative sample counts per label</h3>
<p><strong>Description:</strong> Find the per-group percentile rank of all labels across both neighbour sets, relative to the processing group. An absence is treated as a sample count of zero.</p>
<p><strong>Subroutine:</strong> calc_label_count_quantile_position</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 33%">
<col style="width: 33%">
<col style="width: 33%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>LABEL_COUNT_RANK_PCT</td>
<td>List of percentile ranks for each label’s sample count</td>
<td>1</td>
</tr>
</tbody>
</table>
</section>
<section id="redundancy" class="level3">
<h3 class="anchored" data-anchor-id="redundancy">Redundancy</h3>
<p><strong>Description:</strong> Ratio of labels to samples. Values close to 1 are well sampled while zero means there is no redundancy in the sampling</p>
<p><strong>Subroutine:</strong> calc_redundancy</p>
<p><strong>Reference:</strong> <a href="https://doi.org/10.1111/j.1654-1103.2003.tb02174.x">Garcillan et al.&nbsp;(2003) J Veget. Sci</a></p>
<p><strong>Formula:</strong> <span class="math inline">\(= 1 - \frac{richness}{sum\ of\ the\ sample\ counts}\)</span></p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Grouping metric?</em></th>
<th><em>Minimum number of neighbour sets</em></th>
<th><em>Formula</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>REDUNDANCY_ALL</td>
<td>for both neighbour sets</td>
<td>region grower</td>
<td>1</td>
<td><span class="math inline">\(= 1 - \frac{RICHNESS\_ALL}{ABC3\_SUM\_ALL}\)</span></td>
</tr>
<tr class="even">
<td>REDUNDANCY_SET1</td>
<td>for neighour set 1</td>
<td></td>
<td>1</td>
<td><span class="math inline">\(= 1 - \frac{RICHNESS\_SET1}{ABC3\_SUM\_SET1}\)</span></td>
</tr>
<tr class="odd">
<td>REDUNDANCY_SET2</td>
<td>for neighour set 2</td>
<td></td>
<td>2</td>
<td><span class="math inline">\(= 1 - \frac{RICHNESS\_SET2}{ABC3\_SUM\_SET2}\)</span></td>
</tr>
</tbody>
</table>
</section>
<section id="richness" class="level3">
<h3 class="anchored" data-anchor-id="richness">Richness</h3>
<p><strong>Description:</strong> Count the number of labels in the neighbour sets</p>
<p><strong>Subroutine:</strong> calc_richness</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Grouping metric?</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>RICHNESS_ALL</td>
<td>for both sets of neighbours</td>
<td>region grower</td>
<td>1</td>
</tr>
<tr class="even">
<td>RICHNESS_SET1</td>
<td>for neighbour set 1</td>
<td></td>
<td>1</td>
</tr>
<tr class="odd">
<td>RICHNESS_SET2</td>
<td>for neighbour set 2</td>
<td></td>
<td>2</td>
</tr>
</tbody>
</table>
</section>
<section id="sample-count-lists" class="level3">
<h3 class="anchored" data-anchor-id="sample-count-lists">Sample count lists</h3>
<p><strong>Description:</strong> Lists of sample counts for each label within the neighbour sets. These form the basis of the sample indices.</p>
<p><strong>Subroutine:</strong> calc_local_sample_count_lists</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 33%">
<col style="width: 33%">
<col style="width: 33%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>ABC3_LABELS_ALL</td>
<td>List of labels in both neighbour sets with their sample counts as the values.</td>
<td>2</td>
</tr>
<tr class="even">
<td>ABC3_LABELS_SET1</td>
<td>List of labels in neighbour set 1. Values are the sample counts.</td>
<td>1</td>
</tr>
<tr class="odd">
<td>ABC3_LABELS_SET2</td>
<td>List of labels in neighbour set 2. Values are the sample counts.</td>
<td>2</td>
</tr>
</tbody>
</table>
</section>
<section id="sample-count-quantiles" class="level3">
<h3 class="anchored" data-anchor-id="sample-count-quantiles">Sample count quantiles</h3>
<p><strong>Description:</strong> Quantiles of the sample counts across the neighbour sets.</p>
<p><strong>Subroutine:</strong> calc_local_sample_count_quantiles</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>ABC3_QUANTILES_ALL</td>
<td>List of quantiles for both neighbour sets</td>
<td>2</td>
</tr>
<tr class="even">
<td>ABC3_QUANTILES_SET1</td>
<td>List of quantiles for neighbour set 1</td>
<td>1</td>
</tr>
<tr class="odd">
<td>ABC3_QUANTILES_SET2</td>
<td>List of quantiles for neighbour set 2</td>
<td>2</td>
</tr>
</tbody>
</table>
</section>
<section id="sample-count-summary-stats" class="level3">
<h3 class="anchored" data-anchor-id="sample-count-summary-stats">Sample count summary stats</h3>
<p><strong>Description:</strong> Summary stats of the sample counts across the neighbour sets.</p>
<p><strong>Subroutine:</strong> calc_local_sample_count_stats</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Grouping metric?</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>ABC3_MEAN_ALL</td>
<td>Mean of label sample counts across both element sets.</td>
<td></td>
<td>2</td>
</tr>
<tr class="even">
<td>ABC3_MEAN_SET1</td>
<td>Mean of label sample counts in neighbour set1.</td>
<td></td>
<td>1</td>
</tr>
<tr class="odd">
<td>ABC3_MEAN_SET2</td>
<td>Mean of label sample counts in neighbour set 2.</td>
<td></td>
<td>2</td>
</tr>
<tr class="even">
<td>ABC3_SD_ALL</td>
<td>Standard deviation of label sample counts in both element sets.</td>
<td></td>
<td>2</td>
</tr>
<tr class="odd">
<td>ABC3_SD_SET1</td>
<td>Standard deviation of sample counts in neighbour set 1.</td>
<td></td>
<td>1</td>
</tr>
<tr class="even">
<td>ABC3_SD_SET2</td>
<td>Standard deviation of label sample counts in neighbour set 2.</td>
<td></td>
<td>2</td>
</tr>
<tr class="odd">
<td>ABC3_SUM_ALL</td>
<td>Sum of the label sample counts across both neighbour sets.</td>
<td>region grower</td>
<td>2</td>
</tr>
<tr class="even">
<td>ABC3_SUM_SET1</td>
<td>Sum of the label sample counts across both neighbour sets.</td>
<td></td>
<td>1</td>
</tr>
<tr class="odd">
<td>ABC3_SUM_SET2</td>
<td>Sum of the label sample counts in neighbour set2.</td>
<td></td>
<td>2</td>
</tr>
</tbody>
</table>
</section>
</section>
<section id="matrix" class="level2">
<h2 class="anchored" data-anchor-id="matrix">Matrix</h2>
<section id="compare-dissimilarity-matrix-values" class="level3">
<h3 class="anchored" data-anchor-id="compare-dissimilarity-matrix-values">Compare dissimilarity matrix values</h3>
<p><strong>Description:</strong> Compare the set of labels in one neighbour set with those in another using their matrix values. Labels not in the matrix are ignored. (This calculation assumes a matrix of dissimilarities and uses 0 as identical, so take care).</p>
<p><strong>Subroutine:</strong> calc_compare_dissim_matrix_values</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Grouping metric?</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>MXD_COUNT</td>
<td>Count of comparisons used.</td>
<td></td>
<td>2</td>
</tr>
<tr class="even">
<td>MXD_LIST1</td>
<td>List of the labels used from neighbour set 1 (those in the matrix). The list values are the number of times each label was used in the calculations. This will always be 1 for labels in neighbour set 1.</td>
<td></td>
<td>2</td>
</tr>
<tr class="odd">
<td>MXD_LIST2</td>
<td>List of the labels used from neighbour set 2 (those in the matrix). The list values are the number of times each label was used in the calculations. This will equal the number of labels used from neighbour set 1.</td>
<td></td>
<td>2</td>
</tr>
<tr class="even">
<td>MXD_MEAN</td>
<td>Mean dissimilarity of labels in set 1 to those in set 2.</td>
<td></td>
<td>2</td>
</tr>
<tr class="odd">
<td>MXD_VARIANCE</td>
<td>Variance of the dissimilarity values, set 1 vs set 2.</td>
<td>cluster metric</td>
<td>2</td>
</tr>
</tbody>
</table>
</section>
<section id="matrix-statistics" class="level3">
<h3 class="anchored" data-anchor-id="matrix-statistics">Matrix statistics</h3>
<p><strong>Description:</strong> Calculate summary statistics of matrix elements in the selected matrix for labels found across both neighbour sets. Labels not in the matrix are ignored.</p>
<p><strong>Subroutine:</strong> calc_matrix_stats</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Grouping metric?</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>MX_KURT</td>
<td>Kurtosis</td>
<td></td>
<td>1</td>
</tr>
<tr class="even">
<td>MX_LABELS</td>
<td>List of the matrix labels in the neighbour sets</td>
<td></td>
<td>1</td>
</tr>
<tr class="odd">
<td>MX_MAXVALUE</td>
<td>Maximum value</td>
<td>region grower</td>
<td>1</td>
</tr>
<tr class="even">
<td>MX_MEAN</td>
<td>Mean</td>
<td></td>
<td>1</td>
</tr>
<tr class="odd">
<td>MX_MEDIAN</td>
<td>Median</td>
<td></td>
<td>1</td>
</tr>
<tr class="even">
<td>MX_MINVALUE</td>
<td>Minimum value</td>
<td></td>
<td>1</td>
</tr>
<tr class="odd">
<td>MX_N</td>
<td>Number of samples (matrix elements, not labels)</td>
<td></td>
<td>1</td>
</tr>
<tr class="even">
<td>MX_PCT05</td>
<td>5th percentile value</td>
<td></td>
<td>1</td>
</tr>
<tr class="odd">
<td>MX_PCT25</td>
<td>First quartile (25th percentile)</td>
<td></td>
<td>1</td>
</tr>
<tr class="even">
<td>MX_PCT75</td>
<td>Third quartile (75th percentile)</td>
<td></td>
<td>1</td>
</tr>
<tr class="odd">
<td>MX_PCT95</td>
<td>95th percentile value</td>
<td></td>
<td>1</td>
</tr>
<tr class="even">
<td>MX_RANGE</td>
<td>Range (max-min)</td>
<td></td>
<td>1</td>
</tr>
<tr class="odd">
<td>MX_SD</td>
<td>Standard deviation</td>
<td></td>
<td>1</td>
</tr>
<tr class="even">
<td>MX_SKEW</td>
<td>Skewness</td>
<td></td>
<td>1</td>
</tr>
<tr class="odd">
<td>MX_VALUES</td>
<td>List of the matrix values</td>
<td></td>
<td>1</td>
</tr>
</tbody>
</table>
</section>
<section id="raos-quadratic-entropy-matrix-weighted" class="level3">
<h3 class="anchored" data-anchor-id="raos-quadratic-entropy-matrix-weighted">Rao’s quadratic entropy, matrix weighted</h3>
<p><strong>Description:</strong> Calculate Rao’s quadratic entropy for a matrix weights scheme. BaseData labels not in the matrix are ignored</p>
<p><strong>Subroutine:</strong> calc_mx_rao_qe</p>
<p><strong>Formula:</strong> <span class="math inline">\(= \sum_{i \in L} \sum_{j \in L} d_{ij} p_i p_j\)</span> where <span class="math inline">\(p_i\)</span> and <span class="math inline">\(p_j\)</span> are the sample counts for the i’th and j’th labels, <span class="math inline">\(d_{ij}\)</span> is the matrix value for the pair of labels <span class="math inline">\(ij\)</span> and <span class="math inline">\(L\)</span> is the set of labels across both neighbour sets that occur in the matrix.</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 33%">
<col style="width: 33%">
<col style="width: 33%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>MX_RAO_QE</td>
<td>Matrix weighted quadratic entropy</td>
<td>1</td>
</tr>
<tr class="even">
<td>MX_RAO_TLABELS</td>
<td>List of labels and values used in the MX_RAO_QE calculations</td>
<td>1</td>
</tr>
<tr class="odd">
<td>MX_RAO_TN</td>
<td>Count of comparisons used to calculate MX_RAO_QE</td>
<td>1</td>
</tr>
</tbody>
</table>
</section>
</section>
<section id="numeric-labels" class="level2">
<h2 class="anchored" data-anchor-id="numeric-labels">Numeric Labels</h2>
<section id="numeric-label-data" class="level3">
<h3 class="anchored" data-anchor-id="numeric-label-data">Numeric label data</h3>
<p><strong>Description:</strong> The underlying data used for the numeric labels stats, as an array. For the hash form, use the ABC3_LABELS_ALL index from the ‘Sample count lists’ calculation.</p>
<p><strong>Subroutine:</strong> calc_numeric_label_data</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 33%">
<col style="width: 33%">
<col style="width: 33%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>NUM_DATA_ARRAY</td>
<td>Numeric label data in array form. Multiple occurrences are repeated based on their sample counts.</td>
<td>1</td>
</tr>
</tbody>
</table>
</section>
<section id="numeric-label-dissimilarity" class="level3">
<h3 class="anchored" data-anchor-id="numeric-label-dissimilarity">Numeric label dissimilarity</h3>
<p><strong>Description:</strong> Compare the set of numeric labels in one neighbour set with those in another.</p>
<p><strong>Subroutine:</strong> calc_numeric_label_dissimilarity</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Grouping metric?</em></th>
<th><em>Minimum number of neighbour sets</em></th>
<th><em>Formula</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>NUMD_ABSMEAN</td>
<td>Mean absolute dissimilarity of labels in set 1 to those in set 2.</td>
<td>cluster metric</td>
<td>2</td>
<td><span class="math inline">\(= \frac{\sum_{l_{1i} \in L_1} \sum_{l_{2j} \in L_2} abs (l_{1i} - l_{2j})(w_{1i} \times w_{2j})}{n_1 \times n_2}\)</span> where <span class="math inline">\(L1\)</span> and <span class="math inline">\(L2\)</span> are the labels in neighbour sets 1 and 2 respectively, and <span class="math inline">\(n1\)</span> and <span class="math inline">\(n2\)</span> are the sample counts in neighbour sets 1 and 2</td>
</tr>
<tr class="even">
<td>NUMD_COUNT</td>
<td>Count of comparisons used.</td>
<td></td>
<td>2</td>
<td><span class="math inline">\(= n1 * n2\)</span> where values are as for <span class="math inline">\(NUMD\_ABSMEAN\)</span></td>
</tr>
<tr class="odd">
<td>NUMD_VARIANCE</td>
<td>Variance of the dissimilarity values (mean squared deviation), set 1 vs set 2.</td>
<td>cluster metric</td>
<td>2</td>
<td><span class="math inline">\(= \frac{\sum_{l_{1i} \in L_1} \sum_{l_{2j} \in L_2} (l_{1i} - l_{2j})^2(w_{1i} \times w_{2j})}{n_1 \times n_2}\)</span> where values are as for <span class="math inline">\(NUMD\_ABSMEAN\)</span></td>
</tr>
</tbody>
</table>
</section>
<section id="numeric-label-harmonic-and-geometric-means" class="level3">
<h3 class="anchored" data-anchor-id="numeric-label-harmonic-and-geometric-means">Numeric label harmonic and geometric means</h3>
<p><strong>Description:</strong> Calculate geometric and harmonic means for a set of numeric labels.</p>
<p><strong>Subroutine:</strong> calc_numeric_label_other_means</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>NUM_GMEAN</td>
<td>Geometric mean</td>
<td>1</td>
</tr>
<tr class="even">
<td>NUM_HMEAN</td>
<td>Harmonic mean</td>
<td>1</td>
</tr>
</tbody>
</table>
</section>
<section id="numeric-label-quantiles" class="level3">
<h3 class="anchored" data-anchor-id="numeric-label-quantiles">Numeric label quantiles</h3>
<p><strong>Description:</strong> Calculate quantiles from a set of numeric labels. Weights by samples so multiple occurrences are accounted for.</p>
<p><strong>Subroutine:</strong> calc_numeric_label_quantiles</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>NUM_Q005</td>
<td>5th percentile</td>
<td>1</td>
</tr>
<tr class="even">
<td>NUM_Q010</td>
<td>10th percentile</td>
<td>1</td>
</tr>
<tr class="odd">
<td>NUM_Q015</td>
<td>15th percentile</td>
<td>1</td>
</tr>
<tr class="even">
<td>NUM_Q020</td>
<td>20th percentile</td>
<td>1</td>
</tr>
<tr class="odd">
<td>NUM_Q025</td>
<td>25th percentile</td>
<td>1</td>
</tr>
<tr class="even">
<td>NUM_Q030</td>
<td>30th percentile</td>
<td>1</td>
</tr>
<tr class="odd">
<td>NUM_Q035</td>
<td>35th percentile</td>
<td>1</td>
</tr>
<tr class="even">
<td>NUM_Q040</td>
<td>40th percentile</td>
<td>1</td>
</tr>
<tr class="odd">
<td>NUM_Q045</td>
<td>45th percentile</td>
<td>1</td>
</tr>
<tr class="even">
<td>NUM_Q050</td>
<td>50th percentile</td>
<td>1</td>
</tr>
<tr class="odd">
<td>NUM_Q055</td>
<td>55th percentile</td>
<td>1</td>
</tr>
<tr class="even">
<td>NUM_Q060</td>
<td>60th percentile</td>
<td>1</td>
</tr>
<tr class="odd">
<td>NUM_Q065</td>
<td>65th percentile</td>
<td>1</td>
</tr>
<tr class="even">
<td>NUM_Q070</td>
<td>70th percentile</td>
<td>1</td>
</tr>
<tr class="odd">
<td>NUM_Q075</td>
<td>75th percentile</td>
<td>1</td>
</tr>
<tr class="even">
<td>NUM_Q080</td>
<td>80th percentile</td>
<td>1</td>
</tr>
<tr class="odd">
<td>NUM_Q085</td>
<td>85th percentile</td>
<td>1</td>
</tr>
<tr class="even">
<td>NUM_Q090</td>
<td>90th percentile</td>
<td>1</td>
</tr>
<tr class="odd">
<td>NUM_Q095</td>
<td>95th percentile</td>
<td>1</td>
</tr>
</tbody>
</table>
</section>
<section id="numeric-label-statistics" class="level3">
<h3 class="anchored" data-anchor-id="numeric-label-statistics">Numeric label statistics</h3>
<p><strong>Description:</strong> Calculate summary statistics from a set of numeric labels. Weights by samples so multiple occurrences are accounted for.</p>
<p><strong>Subroutine:</strong> calc_numeric_label_stats</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Grouping metric?</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>NUM_CV</td>
<td>Coefficient of variation (NUM_SD / NUM_MEAN)</td>
<td></td>
<td>1</td>
</tr>
<tr class="even">
<td>NUM_KURT</td>
<td>Kurtosis</td>
<td></td>
<td>1</td>
</tr>
<tr class="odd">
<td>NUM_MAX</td>
<td>Maximum value (100th quantile)</td>
<td>region grower</td>
<td>1</td>
</tr>
<tr class="even">
<td>NUM_MEAN</td>
<td>Mean</td>
<td></td>
<td>1</td>
</tr>
<tr class="odd">
<td>NUM_MIN</td>
<td>Minimum value (zero quantile)</td>
<td></td>
<td>1</td>
</tr>
<tr class="even">
<td>NUM_N</td>
<td>Number of samples</td>
<td>region grower</td>
<td>1</td>
</tr>
<tr class="odd">
<td>NUM_RANGE</td>
<td>Range (max - min)</td>
<td></td>
<td>1</td>
</tr>
<tr class="even">
<td>NUM_SD</td>
<td>Standard deviation</td>
<td></td>
<td>1</td>
</tr>
<tr class="odd">
<td>NUM_SKEW</td>
<td>Skewness</td>
<td></td>
<td>1</td>
</tr>
</tbody>
</table>
</section>
<section id="numeric-labels-gi-statistic" class="level3">
<h3 class="anchored" data-anchor-id="numeric-labels-gi-statistic">Numeric labels Gi* statistic</h3>
<p><strong>Description:</strong> Getis-Ord Gi* statistic for numeric labels across both neighbour sets</p>
<p><strong>Subroutine:</strong> calc_num_labels_gistar</p>
<p><strong>Reference:</strong> <a href="https://doi.org/10.1111/j.1538-4632.1992.tb00261.x">Getis and Ord (1992) Geographical Analysis</a></p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>NUM_GISTAR</td>
<td>List of Gi* scores</td>
<td>1</td>
</tr>
</tbody>
</table>
</section>
</section>
<section id="phylocom-indices" class="level2">
<h2 class="anchored" data-anchor-id="phylocom-indices">PhyloCom Indices</h2>
<section id="nri-and-nti-expected-values" class="level3">
<h3 class="anchored" data-anchor-id="nri-and-nti-expected-values">NRI and NTI expected values</h3>
<p><strong>Description:</strong> Expected values used in the NRI and NTI calculations. Derived using a null model without resampling where each label has an equal probability of being selected (a null model of even distrbution). The expected mean and SD are the same for each unique number of labels across all neighbour sets. This means if you have three neighbour sets, each with three labels, then the expected values will be identical for each, even if the labels are completely different.</p>
<p><strong>Subroutine:</strong> calc_nri_nti_expected_values</p>
<p><strong>Reference:</strong> <a href="https://doi.org/10.1007/978-3-642-33122-0_3">Webb et al.&nbsp;(2008) https://doi.org/10.1093/bioinformatics/btn358, Tsirogiannis et al.&nbsp;(2012)</a></p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 33%">
<col style="width: 33%">
<col style="width: 33%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>PHYLO_NRI_NTI_SAMPLE_N</td>
<td>Number of random resamples used</td>
<td>1</td>
</tr>
<tr class="even">
<td>PHYLO_NRI_SAMPLE_MEAN</td>
<td>Expected mean of pair-wise distances</td>
<td>1</td>
</tr>
<tr class="odd">
<td>PHYLO_NRI_SAMPLE_SD</td>
<td>Expected standard deviation of pair-wise distances</td>
<td>1</td>
</tr>
<tr class="even">
<td>PHYLO_NTI_SAMPLE_MEAN</td>
<td>Expected mean of nearest taxon distances</td>
<td>1</td>
</tr>
<tr class="odd">
<td>PHYLO_NTI_SAMPLE_SD</td>
<td>Expected standard deviation of nearest taxon distances</td>
<td>1</td>
</tr>
</tbody>
</table>
</section>
<section id="nri-and-nti-abundance-weighted" class="level3">
<h3 class="anchored" data-anchor-id="nri-and-nti-abundance-weighted">NRI and NTI, abundance weighted</h3>
<p><strong>Description:</strong> NRI and NTI for the set of labels on the tree in the sample. This version is -1* the Phylocom implementation, so values &gt;0 have longer branches than expected. Abundance weighted.</p>
<p><strong>Subroutine:</strong> calc_nri_nti3</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>PHYLO_NRI3</td>
<td>Net Relatedness Index, abundance weighted</td>
<td>1</td>
</tr>
<tr class="even">
<td>PHYLO_NTI3</td>
<td>Nearest Taxon Index, abundance weighted</td>
<td>1</td>
</tr>
</tbody>
</table>
</section>
<section id="nri-and-nti-local-range-weighted" class="level3">
<h3 class="anchored" data-anchor-id="nri-and-nti-local-range-weighted">NRI and NTI, local range weighted</h3>
<p><strong>Description:</strong> NRI and NTI for the set of labels on the tree in the sample. This version is -1* the Phylocom implementation, so values &gt;0 have longer branches than expected. Local range weighted.</p>
<p><strong>Subroutine:</strong> calc_nri_nti2</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>PHYLO_NRI2</td>
<td>Net Relatedness Index, local range weighted</td>
<td>1</td>
</tr>
<tr class="even">
<td>PHYLO_NTI2</td>
<td>Nearest Taxon Index, local range weighted</td>
<td>1</td>
</tr>
</tbody>
</table>
</section>
<section id="nri-and-nti-unweighted" class="level3">
<h3 class="anchored" data-anchor-id="nri-and-nti-unweighted">NRI and NTI, unweighted</h3>
<p><strong>Description:</strong> NRI and NTI for the set of labels on the tree in the sample. This version is -1* the Phylocom implementation, so values &gt;0 have longer branches than expected. Not weighted by sample counts, so each label counts once only.</p>
<p><strong>Subroutine:</strong> calc_nri_nti1</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
<th><em>Formula</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>PHYLO_NRI1</td>
<td>Net Relatedness Index, unweighted</td>
<td>1</td>
<td><span class="math inline">\(NRI = \frac{MPD_{obs} - mean(MPD_{rand})}{sd(MPD_{rand})}\)</span></td>
</tr>
<tr class="even">
<td>PHYLO_NTI1</td>
<td>Nearest Taxon Index, unweighted</td>
<td>1</td>
<td><span class="math inline">\(NTI = \frac{MNTD_{obs} - mean(MNTD_{rand})}{sd(MNTD_{rand})}\)</span></td>
</tr>
</tbody>
</table>
</section>
<section id="net-vpd-expected-values" class="level3">
<h3 class="anchored" data-anchor-id="net-vpd-expected-values">Net VPD expected values</h3>
<p><strong>Description:</strong> Expected values for VPD, analogous to the NRI/NTI results</p>
<p><strong>Subroutine:</strong> calc_vpd_expected_values</p>
<p><strong>Reference:</strong> <a href="https://dx.doi.org/10.3354/meps216265">Warwick &amp; Clarke (2001)</a></p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 33%">
<col style="width: 33%">
<col style="width: 33%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>PHYLO_NET_VPD_SAMPLE_MEAN</td>
<td>Expected mean of pair-wise variance (VPD)</td>
<td>1</td>
</tr>
<tr class="even">
<td>PHYLO_NET_VPD_SAMPLE_N</td>
<td>Number of random resamples used to calculate expected pair-wise variance scores(will equal PHYLO_NRI_NTI_SAMPLE_N for non-ultrametric trees)</td>
<td>1</td>
</tr>
<tr class="odd">
<td>PHYLO_NET_VPD_SAMPLE_SD</td>
<td>Expected standard deviation of pair-wise variance (VPD)</td>
<td>1</td>
</tr>
</tbody>
</table>
</section>
<section id="net-variance-of-pair-wise-phylogenetic-distances-unweighted" class="level3">
<h3 class="anchored" data-anchor-id="net-variance-of-pair-wise-phylogenetic-distances-unweighted">Net variance of pair-wise phylogenetic distances, unweighted</h3>
<p><strong>Description:</strong> Z-score of VPD calculated using NRI/NTI resampling Not weighted by sample counts, so each label counts once only.</p>
<p><strong>Subroutine:</strong> calc_net_vpd</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 33%">
<col style="width: 33%">
<col style="width: 33%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>PHYLO_NET_VPD</td>
<td>Net variance of pair-wise phylogenetic distances, unweighted</td>
<td>1</td>
</tr>
</tbody>
</table>
</section>
<section id="phylogenetic-and-nearest-taxon-distances-abundance-weighted" class="level3">
<h3 class="anchored" data-anchor-id="phylogenetic-and-nearest-taxon-distances-abundance-weighted">Phylogenetic and Nearest taxon distances, abundance weighted</h3>
<p><strong>Description:</strong> Distance stats from each label to the nearest label along the tree. Compares with all other labels across both neighbour sets. Weighted by sample counts (which currently must be integers)</p>
<p><strong>Subroutine:</strong> calc_phylo_mpd_mntd3</p>
<p><strong>Reference:</strong> <a href="https://doi.org/10.1093/bioinformatics/btn358">Webb et al.&nbsp;(2008)</a></p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Grouping metric?</em></th>
<th><em>Minimum number of neighbour sets</em></th>
<th><em>Formula</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>PMPD3_MAX</td>
<td>Maximum of pairwise phylogenetic distances</td>
<td></td>
<td>1</td>
<td></td>
</tr>
<tr class="even">
<td>PMPD3_MEAN</td>
<td>Mean of pairwise phylogenetic distances</td>
<td></td>
<td>1</td>
<td><span class="math inline">\(MPD = \frac {\sum_{t_i = 1}^{n_t-1} \sum_{t_j = 1}^{n_t} d_{t_i \leftrightarrow t_j}}{(n_t-1)^2}, i \neq j\)</span> where <span class="math inline">\(d_{t_i \leftrightarrow t_j} = \sum_{b \in B_{t_i \leftrightarrow t_j}} L_b\)</span> is the sum of the branch lengths along the path connecting <span class="math inline">\(t_i\)</span> and <span class="math inline">\(t_j\)</span> such that <span class="math inline">\(L_b\)</span> is the length of each branch in the set of branches <span class="math inline">\(B\)</span></td>
</tr>
<tr class="odd">
<td>PMPD3_MIN</td>
<td>Minimum of pairwise phylogenetic distances</td>
<td></td>
<td>1</td>
<td></td>
</tr>
<tr class="even">
<td>PMPD3_N</td>
<td>Count of pairwise phylogenetic distances</td>
<td></td>
<td>1</td>
<td></td>
</tr>
<tr class="odd">
<td>PMPD3_RMSD</td>
<td>Root mean squared pairwise phylogenetic distances</td>
<td></td>
<td>1</td>
<td></td>
</tr>
<tr class="even">
<td>PMPD3_VARIANCE</td>
<td>Variance of pairwise phylogenetic distances, similar to Clarke and Warwick (2001; http://dx.doi.org/10.3354/meps216265) but uses tip-to-tip distances instead of tip to most recent common ancestor.</td>
<td></td>
<td>1</td>
<td></td>
</tr>
<tr class="odd">
<td>PNTD3_MAX</td>
<td>Maximum of nearest taxon distances</td>
<td>region grower</td>
<td>1</td>
<td></td>
</tr>
<tr class="even">
<td>PNTD3_MEAN</td>
<td>Mean of nearest taxon distances</td>
<td></td>
<td>1</td>
<td></td>
</tr>
<tr class="odd">
<td>PNTD3_MIN</td>
<td>Minimum of nearest taxon distances</td>
<td></td>
<td>1</td>
<td></td>
</tr>
<tr class="even">
<td>PNTD3_N</td>
<td>Count of nearest taxon distances</td>
<td></td>
<td>1</td>
<td></td>
</tr>
<tr class="odd">
<td>PNTD3_RMSD</td>
<td>Root mean squared nearest taxon distances</td>
<td></td>
<td>1</td>
<td></td>
</tr>
<tr class="even">
<td>PNTD3_VARIANCE</td>
<td>Variance of nearest taxon distances</td>
<td></td>
<td>1</td>
<td></td>
</tr>
</tbody>
</table>
</section>
<section id="phylogenetic-and-nearest-taxon-distances-local-range-weighted" class="level3">
<h3 class="anchored" data-anchor-id="phylogenetic-and-nearest-taxon-distances-local-range-weighted">Phylogenetic and Nearest taxon distances, local range weighted</h3>
<p><strong>Description:</strong> Distance stats from each label to the nearest label along the tree. Compares with all other labels across both neighbour sets. Weighted by sample counts</p>
<p><strong>Subroutine:</strong> calc_phylo_mpd_mntd2</p>
<p><strong>Reference:</strong> <a href="https://doi.org/10.1093/bioinformatics/btn358">Webb et al.&nbsp;(2008)</a></p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Grouping metric?</em></th>
<th><em>Minimum number of neighbour sets</em></th>
<th><em>Formula</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>PMPD2_MAX</td>
<td>Maximum of pairwise phylogenetic distances</td>
<td></td>
<td>1</td>
<td></td>
</tr>
<tr class="even">
<td>PMPD2_MEAN</td>
<td>Mean of pairwise phylogenetic distances</td>
<td></td>
<td>1</td>
<td><span class="math inline">\(MPD = \frac {\sum_{t_i = 1}^{n_t-1} \sum_{t_j = 1}^{n_t} d_{t_i \leftrightarrow t_j}}{(n_t-1)^2}, i \neq j\)</span> where <span class="math inline">\(d_{t_i \leftrightarrow t_j} = \sum_{b \in B_{t_i \leftrightarrow t_j}} L_b\)</span> is the sum of the branch lengths along the path connecting <span class="math inline">\(t_i\)</span> and <span class="math inline">\(t_j\)</span> such that <span class="math inline">\(L_b\)</span> is the length of each branch in the set of branches <span class="math inline">\(B\)</span></td>
</tr>
<tr class="odd">
<td>PMPD2_MIN</td>
<td>Minimum of pairwise phylogenetic distances</td>
<td></td>
<td>1</td>
<td></td>
</tr>
<tr class="even">
<td>PMPD2_N</td>
<td>Count of pairwise phylogenetic distances</td>
<td></td>
<td>1</td>
<td></td>
</tr>
<tr class="odd">
<td>PMPD2_RMSD</td>
<td>Root mean squared pairwise phylogenetic distances</td>
<td></td>
<td>1</td>
<td></td>
</tr>
<tr class="even">
<td>PMPD2_VARIANCE</td>
<td>Variance of pairwise phylogenetic distances, similar to Clarke and Warwick (2001; http://dx.doi.org/10.3354/meps216265) but uses tip-to-tip distances instead of tip to most recent common ancestor.</td>
<td></td>
<td>1</td>
<td></td>
</tr>
<tr class="odd">
<td>PNTD2_MAX</td>
<td>Maximum of nearest taxon distances</td>
<td>region grower</td>
<td>1</td>
<td></td>
</tr>
<tr class="even">
<td>PNTD2_MEAN</td>
<td>Mean of nearest taxon distances</td>
<td></td>
<td>1</td>
<td></td>
</tr>
<tr class="odd">
<td>PNTD2_MIN</td>
<td>Minimum of nearest taxon distances</td>
<td></td>
<td>1</td>
<td></td>
</tr>
<tr class="even">
<td>PNTD2_N</td>
<td>Count of nearest taxon distances</td>
<td></td>
<td>1</td>
<td></td>
</tr>
<tr class="odd">
<td>PNTD2_RMSD</td>
<td>Root mean squared nearest taxon distances</td>
<td></td>
<td>1</td>
<td></td>
</tr>
<tr class="even">
<td>PNTD2_VARIANCE</td>
<td>Variance of nearest taxon distances</td>
<td></td>
<td>1</td>
<td></td>
</tr>
</tbody>
</table>
</section>
<section id="phylogenetic-and-nearest-taxon-distances-unweighted" class="level3">
<h3 class="anchored" data-anchor-id="phylogenetic-and-nearest-taxon-distances-unweighted">Phylogenetic and Nearest taxon distances, unweighted</h3>
<p><strong>Description:</strong> Distance stats from each label to the nearest label along the tree. Compares with all other labels across both neighbour sets.</p>
<p><strong>Subroutine:</strong> calc_phylo_mpd_mntd1</p>
<p><strong>Reference:</strong> <a href="https://doi.org/10.1093/bioinformatics/btn358">Webb et al.&nbsp;(2008)</a></p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Grouping metric?</em></th>
<th><em>Minimum number of neighbour sets</em></th>
<th><em>Formula</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>PMPD1_MAX</td>
<td>Maximum of pairwise phylogenetic distances</td>
<td></td>
<td>1</td>
<td></td>
</tr>
<tr class="even">
<td>PMPD1_MEAN</td>
<td>Mean of pairwise phylogenetic distances</td>
<td></td>
<td>1</td>
<td><span class="math inline">\(MPD = \frac {\sum_{t_i = 1}^{n_t-1} \sum_{t_j = 1}^{n_t} d_{t_i \leftrightarrow t_j}}{(n_t-1)^2}, i \neq j\)</span> where <span class="math inline">\(d_{t_i \leftrightarrow t_j} = \sum_{b \in B_{t_i \leftrightarrow t_j}} L_b\)</span> is the sum of the branch lengths along the path connecting <span class="math inline">\(t_i\)</span> and <span class="math inline">\(t_j\)</span> such that <span class="math inline">\(L_b\)</span> is the length of each branch in the set of branches <span class="math inline">\(B\)</span></td>
</tr>
<tr class="odd">
<td>PMPD1_MIN</td>
<td>Minimum of pairwise phylogenetic distances</td>
<td></td>
<td>1</td>
<td></td>
</tr>
<tr class="even">
<td>PMPD1_N</td>
<td>Count of pairwise phylogenetic distances</td>
<td></td>
<td>1</td>
<td></td>
</tr>
<tr class="odd">
<td>PMPD1_RMSD</td>
<td>Root mean squared pairwise phylogenetic distances</td>
<td></td>
<td>1</td>
<td></td>
</tr>
<tr class="even">
<td>PMPD1_VARIANCE</td>
<td>Variance of pairwise phylogenetic distances, similar to Clarke and Warwick (2001; http://dx.doi.org/10.3354/meps216265) but uses tip-to-tip distances instead of tip to most recent common ancestor.</td>
<td></td>
<td>1</td>
<td></td>
</tr>
<tr class="odd">
<td>PNTD1_MAX</td>
<td>Maximum of nearest taxon distances</td>
<td>region grower</td>
<td>1</td>
<td></td>
</tr>
<tr class="even">
<td>PNTD1_MEAN</td>
<td>Mean of nearest taxon distances</td>
<td></td>
<td>1</td>
<td></td>
</tr>
<tr class="odd">
<td>PNTD1_MIN</td>
<td>Minimum of nearest taxon distances</td>
<td></td>
<td>1</td>
<td></td>
</tr>
<tr class="even">
<td>PNTD1_N</td>
<td>Count of nearest taxon distances</td>
<td></td>
<td>1</td>
<td></td>
</tr>
<tr class="odd">
<td>PNTD1_RMSD</td>
<td>Root mean squared nearest taxon distances</td>
<td></td>
<td>1</td>
<td></td>
</tr>
<tr class="even">
<td>PNTD1_VARIANCE</td>
<td>Variance of nearest taxon distances</td>
<td></td>
<td>1</td>
<td></td>
</tr>
</tbody>
</table>
</section>
</section>
<section id="phylogenetic-endemism-indices" class="level2">
<h2 class="anchored" data-anchor-id="phylogenetic-endemism-indices">Phylogenetic Endemism Indices</h2>
<section id="corrected-weighted-phylogenetic-endemism" class="level3">
<h3 class="anchored" data-anchor-id="corrected-weighted-phylogenetic-endemism">Corrected weighted phylogenetic endemism</h3>
<p><strong>Description:</strong> What proportion of the PD is range-restricted to this neighbour set?</p>
<p><strong>Subroutine:</strong> calc_phylo_corrected_weighted_endemism</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
<th><em>Formula</em></th>
<th><em>Reference</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>PE_CWE</td>
<td>Corrected weighted endemism. This is the phylogenetic analogue of corrected weighted endemism.</td>
<td>1</td>
<td><span class="math inline">\(PE\_WE / PD\)</span></td>
<td></td>
</tr>
</tbody>
</table>
</section>
<section id="corrected-weighted-phylogenetic-endemism-central-variant" class="level3">
<h3 class="anchored" data-anchor-id="corrected-weighted-phylogenetic-endemism-central-variant">Corrected weighted phylogenetic endemism, central variant</h3>
<p><strong>Description:</strong> What proportion of the PD in neighbour set 1 is range-restricted to neighbour sets 1 and 2?</p>
<p><strong>Subroutine:</strong> calc_pe_central_cwe</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 33%">
<col style="width: 33%">
<col style="width: 33%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>PEC_CWE</td>
<td>Corrected weighted phylogenetic endemism, central variant</td>
<td>1</td>
</tr>
<tr class="even">
<td>PEC_CWE_PD</td>
<td>PD used in the PEC_CWE index.</td>
<td>1</td>
</tr>
</tbody>
</table>
</section>
<section id="corrected-weighted-phylogenetic-rarity" class="level3">
<h3 class="anchored" data-anchor-id="corrected-weighted-phylogenetic-rarity">Corrected weighted phylogenetic rarity</h3>
<p><strong>Description:</strong> What proportion of the PD is abundance-restricted to this neighbour set?</p>
<p><strong>Subroutine:</strong> calc_phylo_corrected_weighted_rarity</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
<th><em>Formula</em></th>
<th><em>Reference</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>PHYLO_RARITY_CWR</td>
<td>Corrected weighted phylogenetic rarity. This is the phylogenetic rarity analogue of corrected weighted endemism.</td>
<td>1</td>
<td><span class="math inline">\(AED_T / PD\)</span></td>
<td></td>
</tr>
</tbody>
</table>
</section>
<section id="pd-endemism" class="level3">
<h3 class="anchored" data-anchor-id="pd-endemism">PD-Endemism</h3>
<p><strong>Description:</strong> Absolute endemism analogue of PE. It is the sum of the branch lengths restricted to the neighbour sets.</p>
<p><strong>Subroutine:</strong> calc_pd_endemism</p>
<p><strong>Reference:</strong> <a href="https://doi.org/10.1111/j.1523-1739.2004.00330.x">See Faith (2004) Cons Biol.</a></p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Grouping metric?</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>PD_ENDEMISM</td>
<td>Phylogenetic Diversity Endemism</td>
<td>region grower</td>
<td>1</td>
</tr>
<tr class="even">
<td>PD_ENDEMISM_P</td>
<td>Phylogenetic Diversity Endemism, as a proportion of the whole tree</td>
<td>region grower</td>
<td>1</td>
</tr>
<tr class="odd">
<td>PD_ENDEMISM_WTS</td>
<td>Phylogenetic Diversity Endemism weights per node found only in the neighbour set</td>
<td></td>
<td>1</td>
</tr>
</tbody>
</table>
</section>
<section id="pe-clade-contributions" class="level3">
<h3 class="anchored" data-anchor-id="pe-clade-contributions">PE clade contributions</h3>
<p><strong>Description:</strong> Contribution of each node and its descendents to the Phylogenetic endemism (PE) calculation.</p>
<p><strong>Subroutine:</strong> calc_pe_clade_contributions</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 33%">
<col style="width: 33%">
<col style="width: 33%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>PE_CLADE_CONTR</td>
<td>List of node (clade) contributions to the PE calculation</td>
<td>1</td>
</tr>
<tr class="even">
<td>PE_CLADE_CONTR_P</td>
<td>List of node (clade) contributions to the PE calculation, proportional to the entire tree</td>
<td>1</td>
</tr>
<tr class="odd">
<td>PE_CLADE_SCORE</td>
<td>List of PE scores for each node (clade), being the sum of all descendent PE weights</td>
<td>1</td>
</tr>
</tbody>
</table>
</section>
<section id="pe-clade-loss" class="level3">
<h3 class="anchored" data-anchor-id="pe-clade-loss">PE clade loss</h3>
<p><strong>Description:</strong> How much of the PE would be lost if a clade were to be removed? Calculates the clade PE below the last ancestral node in the neighbour set which would still be in the neighbour set.</p>
<p><strong>Subroutine:</strong> calc_pe_clade_loss</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 33%">
<col style="width: 33%">
<col style="width: 33%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>PE_CLADE_LOSS_CONTR</td>
<td>List of the proportion of the PE score which would be lost if each clade were removed.</td>
<td>1</td>
</tr>
<tr class="even">
<td>PE_CLADE_LOSS_CONTR_P</td>
<td>As per PE_CLADE_LOSS but proportional to the entire tree</td>
<td>1</td>
</tr>
<tr class="odd">
<td>PE_CLADE_LOSS_SCORE</td>
<td>List of how much PE would be lost if each clade were removed.</td>
<td>1</td>
</tr>
</tbody>
</table>
</section>
<section id="pe-clade-loss-ancestral-component" class="level3">
<h3 class="anchored" data-anchor-id="pe-clade-loss-ancestral-component">PE clade loss (ancestral component)</h3>
<p><strong>Description:</strong> How much of the PE clade loss is due to the ancestral branches? The score is zero when there is no ancestral loss.</p>
<p><strong>Subroutine:</strong> calc_pe_clade_loss_ancestral</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 33%">
<col style="width: 33%">
<col style="width: 33%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>PE_CLADE_LOSS_ANC</td>
<td>List of how much ancestral PE would be lost if each clade were removed. The value is 0 when no ancestral PE is lost.</td>
<td>1</td>
</tr>
<tr class="even">
<td>PE_CLADE_LOSS_ANC_P</td>
<td>List of the proportion of the clade’s PE loss that is due to the ancestral branches.</td>
<td>1</td>
</tr>
</tbody>
</table>
</section>
<section id="phylogenetic-endemism" class="level3">
<h3 class="anchored" data-anchor-id="phylogenetic-endemism">Phylogenetic Endemism</h3>
<p><strong>Description:</strong> Phylogenetic endemism (PE). Uses labels across both neighbourhoods and trims the tree to exclude labels not in the BaseData object.</p>
<p><strong>Subroutine:</strong> calc_pe</p>
<p><strong>Reference:</strong> <a href="https://doi.org/10.1111/j.1365-294X.2009.04311.x">Rosauer et al (2009) Mol. Ecol</a>; <a href="https://doi.org/10.1016/j.scitotenv.2015.04.113">Laity et al.&nbsp;(2015)</a>; <a href="https://doi.org/10.1111/2041-210X.12513">Laffan et al.&nbsp;(2016)</a></p>
<p><strong>Formula:</strong> <span class="math inline">\(PE = \sum_{\lambda \in \Lambda } L_{\lambda}\frac{r_\lambda}{R_\lambda}\)</span> where <span class="math inline">\(\Lambda\)</span> is the set of branches found across neighbour sets 1 and 2, <span class="math inline">\(L_\lambda\)</span> is the length of branch <span class="math inline">\(\lambda\)</span> , <span class="math inline">\(r_\lambda\)</span> is the local range of branch <span class="math inline">\(\lambda\)</span> (the number of groups in neighbour sets 1 and 2 containing it), and <span class="math inline">\(R_\lambda\)</span> is the global range of branch <span class="math inline">\(\lambda\)</span> (the number of groups across the entire data set containing it).</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Grouping metric?</em></th>
<th><em>Minimum number of neighbour sets</em></th>
<th><em>Formula</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>PE_WE</td>
<td>Phylogenetic endemism</td>
<td>region grower</td>
<td>1</td>
<td></td>
</tr>
<tr class="even">
<td>PE_WE_P</td>
<td>Phylogenetic weighted endemism as a proportion of the total tree length</td>
<td>region grower</td>
<td>1</td>
<td><span class="math inline">\(PE\_WE / L\)</span> where L is the sum of all branch lengths in the trimmed tree</td>
</tr>
</tbody>
</table>
</section>
<section id="phylogenetic-endemism-central" class="level3">
<h3 class="anchored" data-anchor-id="phylogenetic-endemism-central">Phylogenetic Endemism central</h3>
<p><strong>Description:</strong> A variant of Phylogenetic endemism (PE) that uses labels from neighbour set 1 but local ranges from across both neighbour sets 1 and 2. Identical to PE if only one neighbour set is specified.</p>
<p><strong>Subroutine:</strong> calc_pe_central</p>
<p><strong>Reference:</strong> <a href="https://doi.org/10.1111/j.1365-294X.2009.04311.x">Rosauer et al (2009) Mol. Ecol</a></p>
<p><strong>Formula:</strong> <span class="math inline">\(PEC = \sum_{\lambda \in \Lambda } L_{\lambda}\frac{r_\lambda}{R_\lambda}\)</span> where <span class="math inline">\(\Lambda\)</span> is the set of branches found across neighbour set 1 only, <span class="math inline">\(L_\lambda\)</span> is the length of branch <span class="math inline">\(\lambda\)</span> , <span class="math inline">\(r_\lambda\)</span> is the local range of branch <span class="math inline">\(\lambda\)</span> (the number of groups in neighbour sets 1 and 2 containing it), and <span class="math inline">\(R_\lambda\)</span> is the global range of branch <span class="math inline">\(\lambda\)</span> (the number of groups across the entire data set containing it).</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Grouping metric?</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>PEC_WE</td>
<td>Phylogenetic endemism, central variant</td>
<td>region grower</td>
<td>1</td>
</tr>
<tr class="even">
<td>PEC_WE_P</td>
<td>Phylogenetic weighted endemism as a proportion of the total tree length, central variant</td>
<td>region grower</td>
<td>1</td>
</tr>
</tbody>
</table>
</section>
<section id="phylogenetic-endemism-central-lists" class="level3">
<h3 class="anchored" data-anchor-id="phylogenetic-endemism-central-lists">Phylogenetic Endemism central lists</h3>
<p><strong>Description:</strong> Lists underlying the phylogenetic endemism central indices. Uses labels from neighbour set one but local ranges from across both neighbour sets.</p>
<p><strong>Subroutine:</strong> calc_pe_central_lists</p>
<p><strong>Reference:</strong> <a href="https://doi.org/10.1111/j.1365-294X.2009.04311.x">Rosauer et al (2009) Mol. Ecol</a></p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 33%">
<col style="width: 33%">
<col style="width: 33%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>PEC_LOCAL_RANGELIST</td>
<td>Phylogenetic endemism local range lists, central variant</td>
<td>1</td>
</tr>
<tr class="even">
<td>PEC_RANGELIST</td>
<td>Phylogenetic endemism global range lists, central variant</td>
<td>1</td>
</tr>
<tr class="odd">
<td>PEC_WTLIST</td>
<td>Phylogenetic endemism weights, central variant</td>
<td>1</td>
</tr>
</tbody>
</table>
</section>
<section id="phylogenetic-endemism-lists" class="level3">
<h3 class="anchored" data-anchor-id="phylogenetic-endemism-lists">Phylogenetic Endemism lists</h3>
<p><strong>Description:</strong> Lists used in the Phylogenetic endemism (PE) calculations.</p>
<p><strong>Subroutine:</strong> calc_pe_lists</p>
<p><strong>Reference:</strong> <a href="https://doi.org/10.1111/j.1365-294X.2009.04311.x">Rosauer et al (2009) Mol. Ecol</a></p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 33%">
<col style="width: 33%">
<col style="width: 33%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>PE_LOCAL_RANGELIST</td>
<td>Local node ranges used in PE calculations (number of groups in which a node is found)</td>
<td>1</td>
</tr>
<tr class="even">
<td>PE_RANGELIST</td>
<td>Node ranges used in PE calculations</td>
<td>1</td>
</tr>
<tr class="odd">
<td>PE_WTLIST</td>
<td>Node weights used in PE calculations</td>
<td>1</td>
</tr>
</tbody>
</table>
</section>
<section id="phylogenetic-endemism-single" class="level3">
<h3 class="anchored" data-anchor-id="phylogenetic-endemism-single">Phylogenetic Endemism single</h3>
<p><strong>Description:</strong> PE scores, but not weighted by local ranges. This is the strict interpretation of the formula given in Rosauer et al.&nbsp;(2009), although the approach has always been implemented as the fraction of each branch’s geographic range that is found in the sample window (see formula for PE_WE).</p>
<p><strong>Subroutine:</strong> calc_pe_single</p>
<p><strong>Reference:</strong> <a href="https://doi.org/10.1111/j.1365-294X.2009.04311.x">Rosauer et al (2009) Mol. Ecol</a></p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Grouping metric?</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>PE_WE_SINGLE</td>
<td>Phylogenetic endemism unweighted by the number of neighbours. Counts each label only once, regardless of how many groups in the neighbourhood it is found in. Useful if your data have sampling biases. Better with small sample windows.</td>
<td>region grower</td>
<td>1</td>
</tr>
<tr class="even">
<td>PE_WE_SINGLE_P</td>
<td>Phylogenetic endemism unweighted by the number of neighbours as a proportion of the total tree length. Counts each label only once, regardless of how many groups in the neighbourhood it is found. Useful if your data have sampling biases.</td>
<td>region grower</td>
<td>1</td>
</tr>
</tbody>
</table>
</section>
<section id="rwibald" class="level3">
<h3 class="anchored" data-anchor-id="rwibald">RWiBaLD</h3>
<p><strong>Description:</strong> Range weighted branch length differences. Values are spatially constant, only the subsets change</p>
<p><strong>Subroutine:</strong> calc_rwibald</p>
<p><strong>Reference:</strong> Mishler et al.&nbsp;(in review)</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 33%">
<col style="width: 33%">
<col style="width: 33%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>RWIBALD_CODES</td>
<td>RWiBaLD codes, 1=palaeo, 2=neo, 3=meso</td>
<td>1</td>
</tr>
<tr class="even">
<td>RWIBALD_CODE_COUNTS</td>
<td>Counts of branches in each RWiBaLD category</td>
<td>1</td>
</tr>
<tr class="odd">
<td>RWIBALD_DIFFS</td>
<td>RWiBaLD scores (continuous differences)</td>
<td>1</td>
</tr>
<tr class="even">
<td>RWIBALD_METADATA</td>
<td>General metadata for the RWiBaLD calculations</td>
<td>1</td>
</tr>
<tr class="odd">
<td>RWIBALD_RR_DIFFS</td>
<td>RWiBaLD scores for the range restricted subset (continuous differences)</td>
<td>1</td>
</tr>
</tbody>
</table>
</section>
</section>
<section id="phylogenetic-indices" class="level2">
<h2 class="anchored" data-anchor-id="phylogenetic-indices">Phylogenetic Indices</h2>
<section id="count-labels-on-tree" class="level3">
<h3 class="anchored" data-anchor-id="count-labels-on-tree">Count labels on tree</h3>
<p><strong>Description:</strong> Count the number of labels that are on the tree</p>
<p><strong>Subroutine:</strong> calc_count_labels_on_tree</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Grouping metric?</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>PHYLO_LABELS_ON_TREE_COUNT</td>
<td>The number of labels that are found on the tree, across both neighbour sets</td>
<td>region grower</td>
<td>1</td>
</tr>
</tbody>
</table>
</section>
<section id="evolutionary-distinctiveness" class="level3">
<h3 class="anchored" data-anchor-id="evolutionary-distinctiveness">Evolutionary distinctiveness</h3>
<p><strong>Description:</strong> Evolutionary distinctiveness metrics (AED, ED, ES) Label values are constant for all neighbourhoods in which each label is found.</p>
<p><strong>Subroutine:</strong> calc_phylo_aed</p>
<p><strong>Reference:</strong> <a href="https://doi.org/10.1111/j.1472-4642.2010.00650.x">Cadotte &amp; Davies (2010)</a></p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
<th><em>Reference</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>PHYLO_AED_LIST</td>
<td>Abundance weighted ED per terminal label</td>
<td>1</td>
<td><a href="https://doi.org/10.1111/j.1472-4642.2010.00650.x">Cadotte &amp; Davies (2010)</a></td>
</tr>
<tr class="even">
<td>PHYLO_ED_LIST</td>
<td>“Fair proportion” partitioning of PD per terminal label</td>
<td>1</td>
<td><a href="https://doi.org/10.1371/journal.pone.0000296">Isaac et al.&nbsp;(2007)</a></td>
</tr>
<tr class="odd">
<td>PHYLO_ES_LIST</td>
<td>Equal splits partitioning of PD per terminal label</td>
<td>1</td>
<td><a href="https://doi.org/10.1111%2Fj.1523-1739.2006.00555.x">Redding &amp; Mooers (2006)</a></td>
</tr>
</tbody>
</table>
</section>
<section id="evolutionary-distinctiveness-per-site" class="level3">
<h3 class="anchored" data-anchor-id="evolutionary-distinctiveness-per-site">Evolutionary distinctiveness per site</h3>
<p><strong>Description:</strong> Site level evolutionary distinctiveness</p>
<p><strong>Subroutine:</strong> calc_phylo_aed_t</p>
<p><strong>Reference:</strong> <a href="https://doi.org/10.1111/j.1472-4642.2010.00650.x">Cadotte &amp; Davies (2010)</a></p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Grouping metric?</em></th>
<th><em>Minimum number of neighbour sets</em></th>
<th><em>Reference</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>PHYLO_AED_T</td>
<td>Abundance weighted ED_t (sum of values in PHYLO_AED_LIST times their abundances). This is equivalent to a phylogenetic rarity score (see phylogenetic endemism)</td>
<td>region grower</td>
<td>1</td>
<td><a href="https://doi.org/10.1111/j.1472-4642.2010.00650.x">Cadotte &amp; Davies (2010)</a></td>
</tr>
</tbody>
</table>
</section>
<section id="evolutionary-distinctiveness-per-terminal-taxon-per-site" class="level3">
<h3 class="anchored" data-anchor-id="evolutionary-distinctiveness-per-terminal-taxon-per-site">Evolutionary distinctiveness per terminal taxon per site</h3>
<p><strong>Description:</strong> Site level evolutionary distinctiveness per terminal taxon</p>
<p><strong>Subroutine:</strong> calc_phylo_aed_t_wtlists</p>
<p><strong>Reference:</strong> <a href="https://doi.org/10.1111/j.1472-4642.2010.00650.x">Cadotte &amp; Davies (2010)</a></p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
<th><em>Reference</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>PHYLO_AED_T_WTLIST</td>
<td>Abundance weighted ED per terminal taxon (the AED score of each taxon multiplied by its abundance in the sample)</td>
<td>1</td>
<td><a href="https://doi.org/10.1111/j.1472-4642.2010.00650.x">Cadotte &amp; Davies (2010)</a></td>
</tr>
<tr class="even">
<td>PHYLO_AED_T_WTLIST_P</td>
<td>Proportional contribution of each terminal taxon to the AED_T score</td>
<td>1</td>
<td><a href="https://doi.org/10.1111/j.1472-4642.2010.00650.x">Cadotte &amp; Davies (2010)</a></td>
</tr>
</tbody>
</table>
</section>
<section id="labels-not-on-tree" class="level3">
<h3 class="anchored" data-anchor-id="labels-not-on-tree">Labels not on tree</h3>
<p><strong>Description:</strong> Create a hash of the labels that are not on the tree</p>
<p><strong>Subroutine:</strong> calc_labels_not_on_tree</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Grouping metric?</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>PHYLO_LABELS_NOT_ON_TREE</td>
<td>A hash of labels that are not found on the tree, across both neighbour sets</td>
<td></td>
<td>1</td>
</tr>
<tr class="even">
<td>PHYLO_LABELS_NOT_ON_TREE_N</td>
<td>Number of labels not on the tree</td>
<td>region grower</td>
<td>1</td>
</tr>
<tr class="odd">
<td>PHYLO_LABELS_NOT_ON_TREE_P</td>
<td>Proportion of labels not on the tree</td>
<td></td>
<td>1</td>
</tr>
</tbody>
</table>
</section>
<section id="labels-on-tree" class="level3">
<h3 class="anchored" data-anchor-id="labels-on-tree">Labels on tree</h3>
<p><strong>Description:</strong> Create a hash of the labels that are on the tree</p>
<p><strong>Subroutine:</strong> calc_labels_on_tree</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 33%">
<col style="width: 33%">
<col style="width: 33%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>PHYLO_LABELS_ON_TREE</td>
<td>A hash of labels that are found on the tree, across both neighbour sets</td>
<td>1</td>
</tr>
</tbody>
</table>
</section>
<section id="last-shared-ancestor-properties" class="level3">
<h3 class="anchored" data-anchor-id="last-shared-ancestor-properties">Last shared ancestor properties</h3>
<p><strong>Description:</strong> Properties of the last shared ancestor of an assemblage. Uses labels in both neighbourhoods.</p>
<p><strong>Subroutine:</strong> calc_last_shared_ancestor_props</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 33%">
<col style="width: 33%">
<col style="width: 33%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>LAST_SHARED_ANCESTOR_DEPTH</td>
<td>Depth of last shared ancestor from the root. The root has a depth of zero.</td>
<td>1</td>
</tr>
<tr class="even">
<td>LAST_SHARED_ANCESTOR_DIST_TO_ROOT</td>
<td>Distance along the tree from the last shared ancestor to the root. Includes the shared ancestor’s length.</td>
<td>1</td>
</tr>
<tr class="odd">
<td>LAST_SHARED_ANCESTOR_DIST_TO_TIP</td>
<td>Distance along the tree from the last shared ancestor to the furthest tip in the sample. This is calculated from the point at which the lineages merge, which is the branch end further from the root</td>
<td>1</td>
</tr>
<tr class="even">
<td>LAST_SHARED_ANCESTOR_LENGTH</td>
<td>Branch length of last shared ancestor</td>
<td>1</td>
</tr>
<tr class="odd">
<td>LAST_SHARED_ANCESTOR_POS_REL</td>
<td>Relative position of the last shared ancestor. Value is the fraction of the distance from the root to the furthest terminal.This uses the point at which the lineages merge, and is the branch end further from the root</td>
<td>1</td>
</tr>
</tbody>
</table>
</section>
<section id="pd-clade-contributions" class="level3">
<h3 class="anchored" data-anchor-id="pd-clade-contributions">PD clade contributions</h3>
<p><strong>Description:</strong> Contribution of each node and its descendents to the Phylogenetic diversity (PD) calculation.</p>
<p><strong>Subroutine:</strong> calc_pd_clade_contributions</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 33%">
<col style="width: 33%">
<col style="width: 33%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>PD_CLADE_CONTR</td>
<td>List of node (clade) contributions to the PD calculation</td>
<td>1</td>
</tr>
<tr class="even">
<td>PD_CLADE_CONTR_P</td>
<td>List of node (clade) contributions to the PD calculation, proportional to the entire tree</td>
<td>1</td>
</tr>
<tr class="odd">
<td>PD_CLADE_SCORE</td>
<td>List of PD scores for each node (clade), being the sum of all descendent branch lengths</td>
<td>1</td>
</tr>
</tbody>
</table>
</section>
<section id="pd-clade-loss" class="level3">
<h3 class="anchored" data-anchor-id="pd-clade-loss">PD clade loss</h3>
<p><strong>Description:</strong> How much of the PD would be lost if a clade were to be removed? Calculates the clade PD below the last ancestral node in the neighbour set which would still be in the neighbour set.</p>
<p><strong>Subroutine:</strong> calc_pd_clade_loss</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 33%">
<col style="width: 33%">
<col style="width: 33%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>PD_CLADE_LOSS_CONTR</td>
<td>List of the proportion of the PD score which would be lost if each clade were removed.</td>
<td>1</td>
</tr>
<tr class="even">
<td>PD_CLADE_LOSS_CONTR_P</td>
<td>As per PD_CLADE_LOSS but proportional to the entire tree</td>
<td>1</td>
</tr>
<tr class="odd">
<td>PD_CLADE_LOSS_SCORE</td>
<td>List of how much PD would be lost if each clade were removed.</td>
<td>1</td>
</tr>
</tbody>
</table>
</section>
<section id="pd-clade-loss-ancestral-component" class="level3">
<h3 class="anchored" data-anchor-id="pd-clade-loss-ancestral-component">PD clade loss (ancestral component)</h3>
<p><strong>Description:</strong> How much of the PD clade loss is due to the ancestral branches? The score is zero when there is no ancestral loss.</p>
<p><strong>Subroutine:</strong> calc_pd_clade_loss_ancestral</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 33%">
<col style="width: 33%">
<col style="width: 33%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>PD_CLADE_LOSS_ANC</td>
<td>List of how much ancestral PE would be lost if each clade were removed. The value is 0 when no ancestral PD is lost.</td>
<td>1</td>
</tr>
<tr class="even">
<td>PD_CLADE_LOSS_ANC_P</td>
<td>List of the proportion of the clade’s PD loss that is due to the ancestral branches.</td>
<td>1</td>
</tr>
</tbody>
</table>
</section>
<section id="phylogenetic-abundance" class="level3">
<h3 class="anchored" data-anchor-id="phylogenetic-abundance">Phylogenetic Abundance</h3>
<p><strong>Description:</strong> Phylogenetic abundance based on branch lengths back to the root of the tree. Uses labels in both neighbourhoods.</p>
<p><strong>Subroutine:</strong> calc_phylo_abundance</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 16%">
<col style="width: 16%">
<col style="width: 16%">
<col style="width: 16%">
<col style="width: 16%">
<col style="width: 16%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Grouping metric?</em></th>
<th><em>Minimum number of neighbour sets</em></th>
<th><em>Formula</em></th>
<th><em>Reference</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>PHYLO_ABUNDANCE</td>
<td>Phylogenetic abundance</td>
<td>region grower</td>
<td>1</td>
<td><span class="math inline">\(= \sum_{c \in C} A \times L_c\)</span> where <span class="math inline">\(C\)</span> is the set of branches in the minimum spanning path joining the labels in both neighbour sets to the root of the tree, <span class="math inline">\(c\)</span> is a branch (a single segment between two nodes) in the spanning path <span class="math inline">\(C\)</span> , and <span class="math inline">\(L_c\)</span> is the length of branch <span class="math inline">\(c\)</span> , and <span class="math inline">\(A\)</span> is the abundance of that branch (the sum of its descendant label abundances).</td>
<td></td>
</tr>
<tr class="even">
<td>PHYLO_ABUNDANCE_BRANCH_HASH</td>
<td>Phylogenetic abundance per branch</td>
<td></td>
<td>1</td>
<td></td>
<td></td>
</tr>
</tbody>
</table>
</section>
<section id="phylogenetic-diversity" class="level3">
<h3 class="anchored" data-anchor-id="phylogenetic-diversity">Phylogenetic Diversity</h3>
<p><strong>Description:</strong> Phylogenetic diversity (PD) based on branch lengths back to the root of the tree. Uses labels in both neighbourhoods.</p>
<p><strong>Subroutine:</strong> calc_pd</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Grouping metric?</em></th>
<th><em>Minimum number of neighbour sets</em></th>
<th><em>Formula</em></th>
<th><em>Reference</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>PD</td>
<td>Phylogenetic diversity</td>
<td>region grower</td>
<td>1</td>
<td><span class="math inline">\(= \sum_{c \in C} L_c\)</span> where <span class="math inline">\(C\)</span> is the set of branches in the minimum spanning path joining the labels in both neighbour sets to the root of the tree, <span class="math inline">\(c\)</span> is a branch (a single segment between two nodes) in the spanning path <span class="math inline">\(C\)</span> , and <span class="math inline">\(L_c\)</span> is the length of branch <span class="math inline">\(c\)</span> .</td>
<td><a href="https://doi.org/10.1016/0006-3207(92)91201-3">Faith (1992) Biol. Cons</a></td>
</tr>
<tr class="even">
<td>PD_P</td>
<td>Phylogenetic diversity as a proportion of total tree length</td>
<td>region grower</td>
<td>1</td>
<td><span class="math inline">\(= \frac { PD }{ \sum_{c \in C} L_c }\)</span> where terms are the same as for PD, but <span class="math inline">\(c\)</span> , <span class="math inline">\(C\)</span> and <span class="math inline">\(L_c\)</span> are calculated for all nodes in the tree.</td>
<td></td>
</tr>
<tr class="odd">
<td>PD_P_per_taxon</td>
<td>Phylogenetic diversity per taxon as a proportion of total tree length</td>
<td></td>
<td>1</td>
<td><span class="math inline">\(= \frac { PD\_P }{ RICHNESS\_ALL }\)</span></td>
<td></td>
</tr>
<tr class="even">
<td>PD_per_taxon</td>
<td>Phylogenetic diversity per taxon</td>
<td></td>
<td>1</td>
<td><span class="math inline">\(= \frac { PD }{ RICHNESS\_ALL }\)</span></td>
<td></td>
</tr>
</tbody>
</table>
</section>
<section id="phylogenetic-diversity-local" class="level3">
<h3 class="anchored" data-anchor-id="phylogenetic-diversity-local">Phylogenetic Diversity (local)</h3>
<p><strong>Description:</strong> Phylogenetic diversity (PD) based on branch lengths back to the last shared ancestor. Uses labels in both neighbourhoods.</p>
<p><strong>Subroutine:</strong> calc_pd_local</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Grouping metric?</em></th>
<th><em>Minimum number of neighbour sets</em></th>
<th><em>Formula</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>PD_LOCAL</td>
<td>Phylogenetic diversity calculated to last shared ancestor</td>
<td>region grower</td>
<td>1</td>
<td><span class="math inline">\(= \sum_{c \in C} L_c\)</span> where <span class="math inline">\(C\)</span> is the set of branches in the minimum spanning path joining the labels in both neighbour sets to the last shared ancestor, <span class="math inline">\(c\)</span> is a branch (a single segment between two nodes) in the spanning path <span class="math inline">\(C\)</span> , and <span class="math inline">\(L_c\)</span> is the length of branch <span class="math inline">\(c\)</span> .</td>
</tr>
<tr class="even">
<td>PD_LOCAL_P</td>
<td>Phylogenetic diversity as a proportion of total tree length</td>
<td>region grower</td>
<td>1</td>
<td><span class="math inline">\(= \frac { PD }{ \sum_{c \in C} L_c }\)</span> where terms are the same as for PD, but <span class="math inline">\(c\)</span> , <span class="math inline">\(C\)</span> and <span class="math inline">\(L_c\)</span> are calculated for all nodes in the tree.</td>
</tr>
</tbody>
</table>
</section>
<section id="phylogenetic-diversity-node-list" class="level3">
<h3 class="anchored" data-anchor-id="phylogenetic-diversity-node-list">Phylogenetic Diversity node list</h3>
<p><strong>Description:</strong> Phylogenetic diversity (PD) nodes used.</p>
<p><strong>Subroutine:</strong> calc_pd_node_list</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 33%">
<col style="width: 33%">
<col style="width: 33%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>PD_INCLUDED_NODE_LIST</td>
<td>List of tree nodes included in the PD calculations</td>
<td>1</td>
</tr>
</tbody>
</table>
</section>
<section id="phylogenetic-diversity-terminal-node-count" class="level3">
<h3 class="anchored" data-anchor-id="phylogenetic-diversity-terminal-node-count">Phylogenetic Diversity terminal node count</h3>
<p><strong>Description:</strong> Number of terminal nodes in neighbour sets 1 and 2.</p>
<p><strong>Subroutine:</strong> calc_pd_terminal_node_count</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 33%">
<col style="width: 33%">
<col style="width: 33%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>PD_INCLUDED_TERMINAL_NODE_COUNT</td>
<td>Count of tree terminal nodes included in the PD calculations</td>
<td>1</td>
</tr>
</tbody>
</table>
</section>
<section id="phylogenetic-diversity-terminal-node-list" class="level3">
<h3 class="anchored" data-anchor-id="phylogenetic-diversity-terminal-node-list">Phylogenetic Diversity terminal node list</h3>
<p><strong>Description:</strong> Phylogenetic diversity (PD) terminal nodes used.</p>
<p><strong>Subroutine:</strong> calc_pd_terminal_node_list</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 33%">
<col style="width: 33%">
<col style="width: 33%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>PD_INCLUDED_TERMINAL_NODE_LIST</td>
<td>List of tree terminal nodes included in the PD calculations</td>
<td>1</td>
</tr>
</tbody>
</table>
</section>
</section>
<section id="phylogenetic-indices-relative" class="level2">
<h2 class="anchored" data-anchor-id="phylogenetic-indices-relative">Phylogenetic Indices (relative)</h2>
<section id="labels-not-on-trimmed-tree" class="level3">
<h3 class="anchored" data-anchor-id="labels-not-on-trimmed-tree">Labels not on trimmed tree</h3>
<p><strong>Description:</strong> Create a hash of the labels that are not on the trimmed tree</p>
<p><strong>Subroutine:</strong> calc_labels_not_on_trimmed_tree</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 33%">
<col style="width: 33%">
<col style="width: 33%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>PHYLO_LABELS_NOT_ON_TRIMMED_TREE</td>
<td>A hash of labels that are not found on the tree after it has been trimmed to the basedata, across both neighbour sets</td>
<td>1</td>
</tr>
<tr class="even">
<td>PHYLO_LABELS_NOT_ON_TRIMMED_TREE_N</td>
<td>Number of labels not on the trimmed tree</td>
<td>1</td>
</tr>
<tr class="odd">
<td>PHYLO_LABELS_NOT_ON_TRIMMED_TREE_P</td>
<td>Proportion of labels not on the trimmed tree</td>
<td>1</td>
</tr>
</tbody>
</table>
</section>
<section id="labels-on-trimmed-tree" class="level3">
<h3 class="anchored" data-anchor-id="labels-on-trimmed-tree">Labels on trimmed tree</h3>
<p><strong>Description:</strong> Create a hash of the labels that are on the trimmed tree</p>
<p><strong>Subroutine:</strong> calc_labels_on_trimmed_tree</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 33%">
<col style="width: 33%">
<col style="width: 33%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>PHYLO_LABELS_ON_TRIMMED_TREE</td>
<td>A hash of labels that are found on the tree after it has been trimmed to match the basedata, across both neighbour sets</td>
<td>1</td>
</tr>
</tbody>
</table>
</section>
<section id="relative-phylogenetic-diversity-type-1" class="level3">
<h3 class="anchored" data-anchor-id="relative-phylogenetic-diversity-type-1">Relative Phylogenetic Diversity, type 1</h3>
<p><strong>Description:</strong> Relative Phylogenetic Diversity type 1 (RPD1). The ratio of the tree’s PD to a null model of PD evenly distributed across terminals and where ancestral nodes are collapsed to zero length.You probably want to use RPD2 instead as it uses the tree’s topology.</p>
<p><strong>Subroutine:</strong> calc_phylo_rpd1</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Grouping metric?</em></th>
<th><em>Minimum number of neighbour sets</em></th>
<th><em>Formula</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>PHYLO_RPD1</td>
<td>RPD1</td>
<td></td>
<td>1</td>
<td></td>
</tr>
<tr class="even">
<td>PHYLO_RPD_DIFF1</td>
<td>How much more or less PD is there than expected, in original tree units.</td>
<td></td>
<td>1</td>
<td><span class="math inline">\(= tree\_length \times (PD\_P - PHYLO\_RPD\_NULL1)\)</span></td>
</tr>
<tr class="odd">
<td>PHYLO_RPD_NULL1</td>
<td>Null model score used as the denominator in the RPD1 calculations</td>
<td>region grower</td>
<td>1</td>
<td></td>
</tr>
</tbody>
</table>
</section>
<section id="relative-phylogenetic-diversity-type-2" class="level3">
<h3 class="anchored" data-anchor-id="relative-phylogenetic-diversity-type-2">Relative Phylogenetic Diversity, type 2</h3>
<p><strong>Description:</strong> Relative Phylogenetic Diversity (RPD), type 2. The ratio of the tree’s PD to a null model of PD evenly distributed across all nodes (all branches are of equal length).</p>
<p><strong>Subroutine:</strong> calc_phylo_rpd2</p>
<p><strong>Reference:</strong> <a href="https://doi.org/10.1038/ncomms5473">Mishler et al.&nbsp;(2014)</a></p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Grouping metric?</em></th>
<th><em>Minimum number of neighbour sets</em></th>
<th><em>Formula</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>PHYLO_RPD2</td>
<td>RPD2</td>
<td></td>
<td>1</td>
<td></td>
</tr>
<tr class="even">
<td>PHYLO_RPD_DIFF2</td>
<td>How much more or less PD is there than expected, in original tree units.</td>
<td></td>
<td>1</td>
<td><span class="math inline">\(= tree\_length \times (PD\_P - PHYLO\_RPD\_NULL2)\)</span></td>
</tr>
<tr class="odd">
<td>PHYLO_RPD_NULL2</td>
<td>Null model score used as the denominator in the RPD2 calculations</td>
<td>region grower</td>
<td>1</td>
<td></td>
</tr>
</tbody>
</table>
</section>
<section id="relative-phylogenetic-endemism-central" class="level3">
<h3 class="anchored" data-anchor-id="relative-phylogenetic-endemism-central">Relative Phylogenetic Endemism, central</h3>
<p><strong>Description:</strong> Relative Phylogenetic Endemism (RPE). The ratio of the tree’s PE to a null model where PE is calculated using a tree where all branches are of equal length. Same as RPE2 except it only uses the branches in the first neighbour set when more than one is set.</p>
<p><strong>Subroutine:</strong> calc_phylo_rpe_central</p>
<p><strong>Reference:</strong> <a href="https://doi.org/10.1038/ncomms5473">Mishler et al.&nbsp;(2014)</a></p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
<th><em>Formula</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>PHYLO_RPEC</td>
<td>Relative Phylogenetic Endemism score, central</td>
<td>1</td>
<td></td>
</tr>
<tr class="even">
<td>PHYLO_RPE_DIFFC</td>
<td>How much more or less PE is there than expected, in original tree units.</td>
<td>1</td>
<td><span class="math inline">\(= tree\_length \times (PE\_WEC\_P - PHYLO\_RPE\_NULLC)\)</span></td>
</tr>
<tr class="odd">
<td>PHYLO_RPE_NULLC</td>
<td>Null score used as the denominator in the PHYLO_RPEC calculations</td>
<td>1</td>
<td></td>
</tr>
</tbody>
</table>
</section>
<section id="relative-phylogenetic-endemism-type-1" class="level3">
<h3 class="anchored" data-anchor-id="relative-phylogenetic-endemism-type-1">Relative Phylogenetic Endemism, type 1</h3>
<p><strong>Description:</strong> Relative Phylogenetic Endemism, type 1 (RPE1). The ratio of the tree’s PE to a null model of PD evenly distributed across terminals, but with the same range per terminal and where ancestral nodes are of zero length (as per RPD1).You probably want to use RPE2 instead as it uses the tree’s topology.</p>
<p><strong>Subroutine:</strong> calc_phylo_rpe1</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Grouping metric?</em></th>
<th><em>Minimum number of neighbour sets</em></th>
<th><em>Formula</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>PHYLO_RPE1</td>
<td>Relative Phylogenetic Endemism score</td>
<td></td>
<td>1</td>
<td></td>
</tr>
<tr class="even">
<td>PHYLO_RPE_DIFF1</td>
<td>How much more or less PE is there than expected, in original tree units.</td>
<td></td>
<td>1</td>
<td><span class="math inline">\(= tree\_length \times (PE\_WE\_P - PHYLO\_RPE\_NULL1)\)</span></td>
</tr>
<tr class="odd">
<td>PHYLO_RPE_NULL1</td>
<td>Null score used as the denominator in the RPE calculations</td>
<td>region grower</td>
<td>1</td>
<td></td>
</tr>
</tbody>
</table>
</section>
<section id="relative-phylogenetic-endemism-type-2" class="level3">
<h3 class="anchored" data-anchor-id="relative-phylogenetic-endemism-type-2">Relative Phylogenetic Endemism, type 2</h3>
<p><strong>Description:</strong> Relative Phylogenetic Endemism (RPE). The ratio of the tree’s PE to a null model where PE is calculated using a tree where all non-zero branches are of equal length.</p>
<p><strong>Subroutine:</strong> calc_phylo_rpe2</p>
<p><strong>Reference:</strong> <a href="https://doi.org/10.1038/ncomms5473">Mishler et al.&nbsp;(2014)</a></p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Grouping metric?</em></th>
<th><em>Minimum number of neighbour sets</em></th>
<th><em>Formula</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>PHYLO_RPE2</td>
<td>Relative Phylogenetic Endemism score, type 2</td>
<td></td>
<td>1</td>
<td></td>
</tr>
<tr class="even">
<td>PHYLO_RPE_DIFF2</td>
<td>How much more or less PE is there than expected, in original tree units.</td>
<td></td>
<td>1</td>
<td><span class="math inline">\(= tree\_length \times (PE\_WE\_P - PHYLO\_RPE\_NULL2)\)</span></td>
</tr>
<tr class="odd">
<td>PHYLO_RPE_NULL2</td>
<td>Null score used as the denominator in the RPE2 calculations</td>
<td>region grower</td>
<td>1</td>
<td></td>
</tr>
</tbody>
</table>
</section>
</section>
<section id="phylogenetic-turnover" class="level2">
<h2 class="anchored" data-anchor-id="phylogenetic-turnover">Phylogenetic Turnover</h2>
<section id="phylo-jaccard" class="level3">
<h3 class="anchored" data-anchor-id="phylo-jaccard">Phylo Jaccard</h3>
<p><strong>Description:</strong> Jaccard phylogenetic dissimilarity between two sets of taxa, represented by spanning sets of branches</p>
<p><strong>Subroutine:</strong> calc_phylo_jaccard</p>
<p><strong>Reference:</strong> <a href="https://doi.org/10.1128/AEM.71.12.8228-8235.2005">Lozupone and Knight (2005)</a></p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Grouping metric?</em></th>
<th><em>Minimum number of neighbour sets</em></th>
<th><em>Formula</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>PHYLO_JACCARD</td>
<td>Phylo Jaccard score</td>
<td>cluster metric</td>
<td>2</td>
<td><span class="math inline">\(= 1 - (A / (A + B + C))\)</span> where A is the length of shared branches, and B and C are the length of branches found only in neighbour sets 1 and 2</td>
</tr>
</tbody>
</table>
</section>
<section id="phylo-range-weighted-turnover" class="level3">
<h3 class="anchored" data-anchor-id="phylo-range-weighted-turnover">Phylo Range weighted Turnover</h3>
<p><strong>Description:</strong> Phylo Range weighted Turnover</p>
<p><strong>Subroutine:</strong> calc_phylo_rw_turnover</p>
<p><strong>Reference:</strong> <a href="https://doi.org/10.1111/2041-210X.12513">Laffan et al.&nbsp;(2016)</a></p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Grouping metric?</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>PHYLO_RW_TURNOVER</td>
<td>Range weighted turnover</td>
<td>cluster metric</td>
<td>2</td>
</tr>
<tr class="even">
<td>PHYLO_RW_TURNOVER_A</td>
<td>Range weighted turnover, shared component</td>
<td>region grower</td>
<td>2</td>
</tr>
<tr class="odd">
<td>PHYLO_RW_TURNOVER_B</td>
<td>Range weighted turnover, component found only in nbr set 1</td>
<td></td>
<td>2</td>
</tr>
<tr class="even">
<td>PHYLO_RW_TURNOVER_C</td>
<td>Range weighted turnover, component found only in nbr set 2</td>
<td></td>
<td>2</td>
</tr>
</tbody>
</table>
</section>
<section id="phylo-s2" class="level3">
<h3 class="anchored" data-anchor-id="phylo-s2">Phylo S2</h3>
<p><strong>Description:</strong> S2 phylogenetic dissimilarity between two sets of taxa, represented by spanning sets of branches</p>
<p><strong>Subroutine:</strong> calc_phylo_s2</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Grouping metric?</em></th>
<th><em>Minimum number of neighbour sets</em></th>
<th><em>Formula</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>PHYLO_S2</td>
<td>Phylo S2 score</td>
<td>cluster metric</td>
<td>2</td>
<td><span class="math inline">\(= 1 - (A / (A + min (B, C)))\)</span> where A is the sum of shared branch lengths, and B and C are the sum of branch lengths foundonly in neighbour sets 1 and 2</td>
</tr>
</tbody>
</table>
</section>
<section id="phylo-sorenson" class="level3">
<h3 class="anchored" data-anchor-id="phylo-sorenson">Phylo Sorenson</h3>
<p><strong>Description:</strong> Sorenson phylogenetic dissimilarity between two sets of taxa, represented by spanning sets of branches</p>
<p><strong>Subroutine:</strong> calc_phylo_sorenson</p>
<p><strong>Reference:</strong> <a href="https://doi.org/10.1073/pnas.0801920105">Bryant et al.&nbsp;(2008)</a></p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Grouping metric?</em></th>
<th><em>Minimum number of neighbour sets</em></th>
<th><em>Formula</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>PHYLO_SORENSON</td>
<td>Phylo Sorenson score</td>
<td>cluster metric</td>
<td>2</td>
<td><span class="math inline">\(1 - (2A / (2A + B + C))\)</span> where A is the length of shared branches, and B and C are the length of branches found only in neighbour sets 1 and 2</td>
</tr>
</tbody>
</table>
</section>
<section id="phylogenetic-abc" class="level3">
<h3 class="anchored" data-anchor-id="phylogenetic-abc">Phylogenetic ABC</h3>
<p><strong>Description:</strong> Calculate the shared and not shared branch lengths between two sets of labels</p>
<p><strong>Subroutine:</strong> calc_phylo_abc</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Grouping metric?</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>PHYLO_A</td>
<td>Sum of branch lengths shared by labels in nbr sets 1 and 2</td>
<td>region grower</td>
<td>2</td>
</tr>
<tr class="even">
<td>PHYLO_ABC</td>
<td>Sum of branch lengths associated with labels in nbr sets 1 and 2</td>
<td>region grower</td>
<td>2</td>
</tr>
<tr class="odd">
<td>PHYLO_B</td>
<td>Sum of branch lengths unique to labels in nbr set 1</td>
<td></td>
<td>2</td>
</tr>
<tr class="even">
<td>PHYLO_C</td>
<td>Sum of branch lengths unique to labels in nbr set 2</td>
<td></td>
<td>2</td>
</tr>
</tbody>
</table>
</section>
</section>
<section id="rarity" class="level2">
<h2 class="anchored" data-anchor-id="rarity">Rarity</h2>
<section id="rarity-central" class="level3">
<h3 class="anchored" data-anchor-id="rarity-central">Rarity central</h3>
<p><strong>Description:</strong> Calculate rarity for species only in neighbour set 1, but with local sample counts calculated from both neighbour sets. Uses the same algorithm as the endemism indices but weights by sample counts instead of by groups occupied.</p>
<p><strong>Subroutine:</strong> calc_rarity_central</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
<th><em>Formula</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>RAREC_CWE</td>
<td>Corrected weighted rarity</td>
<td>1</td>
<td><span class="math inline">\(= \frac{RAREC\_WE}{RAREC\_RICHNESS}\)</span></td>
</tr>
<tr class="even">
<td>RAREC_RICHNESS</td>
<td>Richness used in RAREC_CWE (same as index RICHNESS_SET1).</td>
<td>1</td>
<td></td>
</tr>
<tr class="odd">
<td>RAREC_WE</td>
<td>Weighted rarity</td>
<td>1</td>
<td><span class="math inline">\(= \sum_{t \in T} \frac {s_t} {S_t}\)</span> where <span class="math inline">\(t\)</span> is a label (taxon) in the set of labels (taxa) <span class="math inline">\(T\)</span> across neighbour set 1, <span class="math inline">\(s_t\)</span> is sum of the sample counts for <span class="math inline">\(t\)</span> across the elements in neighbour sets 1 &amp; 2 (its value in list ABC3_LABELS_ALL), and <span class="math inline">\(S_t\)</span> is the total number of samples across the data set for label <span class="math inline">\(t\)</span> (unless the total sample count is specified at import).</td>
</tr>
</tbody>
</table>
</section>
<section id="rarity-central-lists" class="level3">
<h3 class="anchored" data-anchor-id="rarity-central-lists">Rarity central lists</h3>
<p><strong>Description:</strong> Lists used in rarity central calculations</p>
<p><strong>Subroutine:</strong> calc_rarity_central_lists</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 33%">
<col style="width: 33%">
<col style="width: 33%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>RAREC_RANGELIST</td>
<td>List of ranges for each label used in the rarity central calculations</td>
<td>1</td>
</tr>
<tr class="even">
<td>RAREC_WTLIST</td>
<td>List of weights for each label used in therarity central calculations</td>
<td>1</td>
</tr>
</tbody>
</table>
</section>
<section id="rarity-whole" class="level3">
<h3 class="anchored" data-anchor-id="rarity-whole">Rarity whole</h3>
<p><strong>Description:</strong> Calculate rarity using all species in both neighbour sets. Uses the same algorithm as the endemism indices but weights by sample counts instead of by groups occupied.</p>
<p><strong>Subroutine:</strong> calc_rarity_whole</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Grouping metric?</em></th>
<th><em>Minimum number of neighbour sets</em></th>
<th><em>Formula</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>RAREW_CWE</td>
<td>Corrected weighted rarity</td>
<td></td>
<td>1</td>
<td><span class="math inline">\(= \frac{RAREW\_WE}{RAREW\_RICHNESS}\)</span></td>
</tr>
<tr class="even">
<td>RAREW_RICHNESS</td>
<td>Richness used in RAREW_CWE (same as index RICHNESS_ALL).</td>
<td>region grower</td>
<td>1</td>
<td></td>
</tr>
<tr class="odd">
<td>RAREW_WE</td>
<td>Weighted rarity</td>
<td>region grower</td>
<td>1</td>
<td><span class="math inline">\(= \sum_{t \in T} \frac {s_t} {S_t}\)</span> where <span class="math inline">\(t\)</span> is a label (taxon) in the set of labels (taxa) <span class="math inline">\(T\)</span> across both neighbour sets, <span class="math inline">\(s_t\)</span> is sum of the sample counts for <span class="math inline">\(t\)</span> across the elements in neighbour sets 1 &amp; 2 (its value in list ABC3_LABELS_ALL), and <span class="math inline">\(S_t\)</span> is the total number of samples across the data set for label <span class="math inline">\(t\)</span> (unless the total sample count is specified at import).</td>
</tr>
</tbody>
</table>
</section>
<section id="rarity-whole-lists" class="level3">
<h3 class="anchored" data-anchor-id="rarity-whole-lists">Rarity whole lists</h3>
<p><strong>Description:</strong> Lists used in rarity whole calculations</p>
<p><strong>Subroutine:</strong> calc_rarity_whole_lists</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 33%">
<col style="width: 33%">
<col style="width: 33%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>RAREW_RANGELIST</td>
<td>List of ranges for each label used in the rarity whole calculations</td>
<td>1</td>
</tr>
<tr class="even">
<td>RAREW_WTLIST</td>
<td>List of weights for each label used in therarity whole calculations</td>
<td>1</td>
</tr>
</tbody>
</table>
</section>
</section>
<section id="richness-estimators" class="level2">
<h2 class="anchored" data-anchor-id="richness-estimators">Richness estimators</h2>
<section id="ace" class="level3">
<h3 class="anchored" data-anchor-id="ace">ACE</h3>
<p><strong>Description:</strong> Abundance Coverage-based Estimator of species richness</p>
<p><strong>Subroutine:</strong> calc_ace</p>
<p><strong>Reference:</strong> <a href="https://doi.org/10.1080/01621459.1992.10475194">Chao and Lee (1992)</a></p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Grouping metric?</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>ACE_CI_LOWER</td>
<td>ACE lower confidence interval estimate</td>
<td>region grower</td>
<td>1</td>
</tr>
<tr class="even">
<td>ACE_CI_UPPER</td>
<td>ACE upper confidence interval estimate</td>
<td>region grower</td>
<td>1</td>
</tr>
<tr class="odd">
<td>ACE_ESTIMATE</td>
<td>ACE score</td>
<td>region grower</td>
<td>1</td>
</tr>
<tr class="even">
<td>ACE_ESTIMATE_USED_CHAO</td>
<td>Set to 1 when ACE cannot be calculated and so Chao1 estimate is used</td>
<td></td>
<td>1</td>
</tr>
<tr class="odd">
<td>ACE_INFREQUENT_COUNT</td>
<td>Count of infrequent species</td>
<td>region grower</td>
<td>1</td>
</tr>
<tr class="even">
<td>ACE_SE</td>
<td>ACE standard error</td>
<td></td>
<td>1</td>
</tr>
<tr class="odd">
<td>ACE_UNDETECTED</td>
<td>Estimated number of undetected species</td>
<td>region grower</td>
<td>1</td>
</tr>
<tr class="even">
<td>ACE_VARIANCE</td>
<td>ACE variance</td>
<td></td>
<td>1</td>
</tr>
</tbody>
</table>
</section>
<section id="chao1" class="level3">
<h3 class="anchored" data-anchor-id="chao1">Chao1</h3>
<p><strong>Description:</strong> Chao1 species richness estimator (abundance based)</p>
<p><strong>Subroutine:</strong> calc_chao1</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Grouping metric?</em></th>
<th><em>Minimum number of neighbour sets</em></th>
<th><em>Reference</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>CHAO1_CI_LOWER</td>
<td>Lower confidence interval for the Chao1 estimate</td>
<td>region grower</td>
<td>1</td>
<td></td>
</tr>
<tr class="even">
<td>CHAO1_CI_UPPER</td>
<td>Upper confidence interval for the Chao1 estimate</td>
<td>region grower</td>
<td>1</td>
<td></td>
</tr>
<tr class="odd">
<td>CHAO1_ESTIMATE</td>
<td>Chao1 index</td>
<td>region grower</td>
<td>1</td>
<td>NEEDED</td>
</tr>
<tr class="even">
<td>CHAO1_F1_COUNT</td>
<td>Number of singletons in the sample</td>
<td>region grower</td>
<td>1</td>
<td></td>
</tr>
<tr class="odd">
<td>CHAO1_F2_COUNT</td>
<td>Number of doubletons in the sample</td>
<td>region grower</td>
<td>1</td>
<td></td>
</tr>
<tr class="even">
<td>CHAO1_META</td>
<td>Metadata indicating which formulae were used in the calculations. Numbers refer to EstimateS equations at https://www.robertkcolwell.org/media_files/63</td>
<td></td>
<td>1</td>
<td></td>
</tr>
<tr class="odd">
<td>CHAO1_SE</td>
<td>Standard error of the Chao1 estimator [= sqrt(variance)]</td>
<td>region grower</td>
<td>1</td>
<td></td>
</tr>
<tr class="even">
<td>CHAO1_UNDETECTED</td>
<td>Estimated number of undetected species</td>
<td>region grower</td>
<td>1</td>
<td></td>
</tr>
<tr class="odd">
<td>CHAO1_VARIANCE</td>
<td>Variance of the Chao1 estimator</td>
<td>region grower</td>
<td>1</td>
<td></td>
</tr>
</tbody>
</table>
</section>
<section id="chao2" class="level3">
<h3 class="anchored" data-anchor-id="chao2">Chao2</h3>
<p><strong>Description:</strong> Chao2 species richness estimator (incidence based)</p>
<p><strong>Subroutine:</strong> calc_chao2</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Grouping metric?</em></th>
<th><em>Minimum number of neighbour sets</em></th>
<th><em>Reference</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>CHAO2_CI_LOWER</td>
<td>Lower confidence interval for the Chao2 estimate</td>
<td>region grower</td>
<td>1</td>
<td></td>
</tr>
<tr class="even">
<td>CHAO2_CI_UPPER</td>
<td>Upper confidence interval for the Chao2 estimate</td>
<td>region grower</td>
<td>1</td>
<td></td>
</tr>
<tr class="odd">
<td>CHAO2_ESTIMATE</td>
<td>Chao2 index</td>
<td>region grower</td>
<td>1</td>
<td>NEEDED</td>
</tr>
<tr class="even">
<td>CHAO2_META</td>
<td>Metadata indicating which formulae were used in the calculations. Numbers refer to EstimateS equations at https://www.robertkcolwell.org/media_files/63</td>
<td></td>
<td>1</td>
<td></td>
</tr>
<tr class="odd">
<td>CHAO2_Q1_COUNT</td>
<td>Number of uniques in the sample</td>
<td>region grower</td>
<td>1</td>
<td></td>
</tr>
<tr class="even">
<td>CHAO2_Q2_COUNT</td>
<td>Number of duplicates in the sample</td>
<td>region grower</td>
<td>1</td>
<td></td>
</tr>
<tr class="odd">
<td>CHAO2_SE</td>
<td>Standard error of the Chao2 estimator [= sqrt (variance)]</td>
<td>region grower</td>
<td>1</td>
<td></td>
</tr>
<tr class="even">
<td>CHAO2_UNDETECTED</td>
<td>Estimated number of undetected species</td>
<td>region grower</td>
<td>1</td>
<td></td>
</tr>
<tr class="odd">
<td>CHAO2_VARIANCE</td>
<td>Variance of the Chao2 estimator</td>
<td>region grower</td>
<td>1</td>
<td></td>
</tr>
</tbody>
</table>
</section>
<section id="hurlbert-richness-estimation" class="level3">
<h3 class="anchored" data-anchor-id="hurlbert-richness-estimation">Hurlbert richness estimation</h3>
<p><strong>Description:</strong> Hurlbert estimated species richness scores for given number of samples.</p>
<p><strong>Subroutine:</strong> calc_hurlbert_es</p>
<p><strong>Reference:</strong> <a href="https://doi.org/10.2307/1934145">Hurlbert, S.H. (1971)</a></p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 33%">
<col style="width: 33%">
<col style="width: 33%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>HURLBERT_ES</td>
<td>List of Hurlbert estimated species richness scores for given number of samples</td>
<td>1</td>
</tr>
</tbody>
</table>
</section>
<section id="ice" class="level3">
<h3 class="anchored" data-anchor-id="ice">ICE</h3>
<p><strong>Description:</strong> Incidence Coverage-based Estimator of species richness</p>
<p><strong>Subroutine:</strong> calc_ice</p>
<p><strong>Reference:</strong> <a href="https://doi.org/10.1016/b978-0-12-384719-5.00424-x">Gotelli and Chao (2013)</a></p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Grouping metric?</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>ICE_CI_LOWER</td>
<td>ICE lower confidence interval estimate</td>
<td>region grower</td>
<td>1</td>
</tr>
<tr class="even">
<td>ICE_CI_UPPER</td>
<td>ICE upper confidence interval estimate</td>
<td>region grower</td>
<td>1</td>
</tr>
<tr class="odd">
<td>ICE_ESTIMATE</td>
<td>ICE score</td>
<td>region grower</td>
<td>1</td>
</tr>
<tr class="even">
<td>ICE_ESTIMATE_USED_CHAO</td>
<td>Set to 1 when ICE cannot be calculated and so Chao2 estimate is used</td>
<td></td>
<td>1</td>
</tr>
<tr class="odd">
<td>ICE_INFREQUENT_COUNT</td>
<td>Count of infrequent species</td>
<td>region grower</td>
<td>1</td>
</tr>
<tr class="even">
<td>ICE_SE</td>
<td>ICE standard error</td>
<td></td>
<td>1</td>
</tr>
<tr class="odd">
<td>ICE_UNDETECTED</td>
<td>Estimated number of undetected species</td>
<td>region grower</td>
<td>1</td>
</tr>
<tr class="even">
<td>ICE_VARIANCE</td>
<td>ICE variance</td>
<td></td>
<td>1</td>
</tr>
</tbody>
</table>
</section>
</section>
<section id="taxonomic-dissimilarity-and-comparison" class="level2">
<h2 class="anchored" data-anchor-id="taxonomic-dissimilarity-and-comparison">Taxonomic Dissimilarity and Comparison</h2>
<section id="beta-diversity" class="level3">
<h3 class="anchored" data-anchor-id="beta-diversity">Beta diversity</h3>
<p><strong>Description:</strong> Beta diversity between neighbour sets 1 and 2.</p>
<p><strong>Subroutine:</strong> calc_beta_diversity</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Grouping metric?</em></th>
<th><em>Minimum number of neighbour sets</em></th>
<th><em>Formula</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>BETA_2</td>
<td>The other beta</td>
<td>cluster metric</td>
<td>2</td>
<td><span class="math inline">\(= \frac{A + B + C}{max((A+B), (A+C))} - 1\)</span> where <span class="math inline">\(A\)</span> is the count of labels found in both neighbour sets, <span class="math inline">\(B\)</span> is the count unique to neighbour set 1, and <span class="math inline">\(C\)</span> is the count unique to neighbour set 2. Use the <a href="#label-counts">Label counts</a> calculation to derive these directly.</td>
</tr>
</tbody>
</table>
</section>
<section id="bray-curtis-non-metric" class="level3">
<h3 class="anchored" data-anchor-id="bray-curtis-non-metric">Bray-Curtis non-metric</h3>
<p><strong>Description:</strong> Bray-Curtis dissimilarity between two sets of labels. Reduces to the Sorenson metric for binary data (where sample counts are 1 or 0).</p>
<p><strong>Subroutine:</strong> calc_bray_curtis</p>
<p><strong>Formula:</strong> <span class="math inline">\(= 1 - \frac{2W}{A + B}\)</span> where <span class="math inline">\(A\)</span> is the sum of the sample counts in neighbour set 1, <span class="math inline">\(B\)</span> is the sum of sample counts in neighbour set 2, and <span class="math inline">\(W=\sum^n_{i=1} min(sample\_count\_label_{i_{set1}},sample\_count\_label_{i_{set2}})\)</span> (meaning it sums the minimum of the sample counts for each of the <span class="math inline">\(n\)</span> labels across the two neighbour sets),</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Grouping metric?</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>BC_A</td>
<td>The A factor used in calculations (see formula)</td>
<td></td>
<td>2</td>
</tr>
<tr class="even">
<td>BC_B</td>
<td>The B factor used in calculations (see formula)</td>
<td></td>
<td>2</td>
</tr>
<tr class="odd">
<td>BC_W</td>
<td>The W factor used in calculations (see formula)</td>
<td>region grower</td>
<td>2</td>
</tr>
<tr class="even">
<td>BRAY_CURTIS</td>
<td>Bray Curtis dissimilarity</td>
<td>cluster metric</td>
<td>2</td>
</tr>
</tbody>
</table>
</section>
<section id="bray-curtis-non-metric-group-count-normalised" class="level3">
<h3 class="anchored" data-anchor-id="bray-curtis-non-metric-group-count-normalised">Bray-Curtis non-metric, group count normalised</h3>
<p><strong>Description:</strong> Bray-Curtis dissimilarity between two neighbourhoods, where the counts in each neighbourhood are divided by the number of groups in each neighbourhood to correct for unbalanced sizes.</p>
<p><strong>Subroutine:</strong> calc_bray_curtis_norm_by_gp_counts</p>
<p><strong>Formula:</strong> <span class="math inline">\(= 1 - \frac{2W}{A + B}\)</span> where <span class="math inline">\(A\)</span> is the sum of the sample counts in neighbour set 1 normalised (divided) by the number of groups, <span class="math inline">\(B\)</span> is the sum of the sample counts in neighbour set 2 normalised by the number of groups, and <span class="math inline">\(W = \sum^n_{i=1} min(sample\_count\_label_{i_{set1}},sample\_count\_label_{i_{set2}})\)</span> (meaning it sums the minimum of the normalised sample counts for each of the <span class="math inline">\(n\)</span> labels across the two neighbour sets),</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Grouping metric?</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>BCN_A</td>
<td>The A factor used in calculations (see formula)</td>
<td></td>
<td>2</td>
</tr>
<tr class="even">
<td>BCN_B</td>
<td>The B factor used in calculations (see formula)</td>
<td></td>
<td>2</td>
</tr>
<tr class="odd">
<td>BCN_W</td>
<td>The W factor used in calculations (see formula)</td>
<td>region grower</td>
<td>2</td>
</tr>
<tr class="even">
<td>BRAY_CURTIS_NORM</td>
<td>Bray Curtis dissimilarity normalised by groups</td>
<td>cluster metric</td>
<td>2</td>
</tr>
</tbody>
</table>
</section>
<section id="jaccard" class="level3">
<h3 class="anchored" data-anchor-id="jaccard">Jaccard</h3>
<p><strong>Description:</strong> Jaccard dissimilarity between the labels in neighbour sets 1 and 2.</p>
<p><strong>Subroutine:</strong> calc_jaccard</p>
<p><strong>Formula:</strong> <span class="math inline">\(= 1 - \frac{A}{A + B + C}\)</span> where <span class="math inline">\(A\)</span> is the count of labels found in both neighbour sets, <span class="math inline">\(B\)</span> is the count unique to neighbour set 1, and <span class="math inline">\(C\)</span> is the count unique to neighbour set 2. Use the <a href="#label-counts">Label counts</a> calculation to derive these directly.</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Grouping metric?</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>JACCARD</td>
<td>Jaccard value, 0 is identical, 1 is completely dissimilar</td>
<td>cluster metric</td>
<td>2</td>
</tr>
</tbody>
</table>
</section>
<section id="kulczynski-2" class="level3">
<h3 class="anchored" data-anchor-id="kulczynski-2">Kulczynski 2</h3>
<p><strong>Description:</strong> Kulczynski 2 dissimilarity between two sets of labels.</p>
<p><strong>Subroutine:</strong> calc_kulczynski2</p>
<p><strong>Formula:</strong> <span class="math inline">\(= 1 - 0.5 \times (\frac{A}{A + B} + \frac{A}{A + C})\)</span> where <span class="math inline">\(A\)</span> is the count of labels found in both neighbour sets, <span class="math inline">\(B\)</span> is the count unique to neighbour set 1, and <span class="math inline">\(C\)</span> is the count unique to neighbour set 2. Use the <a href="#label-counts">Label counts</a> calculation to derive these directly.</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Grouping metric?</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>KULCZYNSKI2</td>
<td>Kulczynski 2 index</td>
<td>cluster metric</td>
<td>2</td>
</tr>
</tbody>
</table>
</section>
<section id="nestedness-resultant" class="level3">
<h3 class="anchored" data-anchor-id="nestedness-resultant">Nestedness-resultant</h3>
<p><strong>Description:</strong> Nestedness-resultant index between the labels in neighbour sets 1 and 2.</p>
<p><strong>Subroutine:</strong> calc_nestedness_resultant</p>
<p><strong>Reference:</strong> <a href="https://doi.org/10.1111/j.1466-8238.2009.00490.x">Baselga (2010) Glob Ecol Biogeog.</a></p>
<p><strong>Formula:</strong> <span class="math inline">\(=\frac{ \left | B - C \right | }{ 2A + B + C } \times \frac { A }{ A + min (B, C) }= SORENSON - S2\)</span> where <span class="math inline">\(A\)</span> is the count of labels found in both neighbour sets, <span class="math inline">\(B\)</span> is the count unique to neighbour set 1, and <span class="math inline">\(C\)</span> is the count unique to neighbour set 2. Use the <a href="#label-counts">Label counts</a> calculation to derive these directly.</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Grouping metric?</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>NEST_RESULTANT</td>
<td>Nestedness-resultant index</td>
<td>cluster metric</td>
<td>2</td>
</tr>
</tbody>
</table>
</section>
<section id="range-weighted-sorenson" class="level3">
<h3 class="anchored" data-anchor-id="range-weighted-sorenson">Range weighted Sorenson</h3>
<p><strong>Description:</strong> Range weighted Sorenson</p>
<p><strong>Subroutine:</strong> calc_rw_turnover</p>
<p><strong>Reference:</strong> <a href="https://doi.org/10.1111/2041-210X.12513">Laffan et al.&nbsp;(2016)</a></p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Grouping metric?</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>RW_TURNOVER</td>
<td>Range weighted turnover</td>
<td>cluster metric</td>
<td>2</td>
</tr>
<tr class="even">
<td>RW_TURNOVER_A</td>
<td>Range weighted turnover, shared component</td>
<td></td>
<td>2</td>
</tr>
<tr class="odd">
<td>RW_TURNOVER_B</td>
<td>Range weighted turnover, component found only in nbr set 1</td>
<td></td>
<td>2</td>
</tr>
<tr class="even">
<td>RW_TURNOVER_C</td>
<td>Range weighted turnover, component found only in nbr set 2</td>
<td></td>
<td>2</td>
</tr>
</tbody>
</table>
</section>
<section id="raos-quadratic-entropy-taxonomically-weighted" class="level3">
<h3 class="anchored" data-anchor-id="raos-quadratic-entropy-taxonomically-weighted">Rao’s quadratic entropy, taxonomically weighted</h3>
<p><strong>Description:</strong> Calculate Rao’s quadratic entropy for a taxonomic weights scheme. Should collapse to be the Simpson index for presence/absence data.</p>
<p><strong>Subroutine:</strong> calc_tx_rao_qe</p>
<p><strong>Formula:</strong> <span class="math inline">\(= \sum_{i \in L} \sum_{j \in L} d_{ij} p_i p_j\)</span> where <span class="math inline">\(p_i\)</span> and <span class="math inline">\(p_j\)</span> are the sample counts for the i’th and j’th labels, <span class="math inline">\(d_{ij}\)</span> is a value of zero if <span class="math inline">\(i = j\)</span> , and a value of 1 otherwise. <span class="math inline">\(L\)</span> is the set of labels across both neighbour sets.</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 33%">
<col style="width: 33%">
<col style="width: 33%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>TX_RAO_QE</td>
<td>Taxonomically weighted quadratic entropy</td>
<td>1</td>
</tr>
<tr class="even">
<td>TX_RAO_TLABELS</td>
<td>List of labels and values used in the TX_RAO_QE calculations</td>
<td>1</td>
</tr>
<tr class="odd">
<td>TX_RAO_TN</td>
<td>Count of comparisons used to calculate TX_RAO_QE</td>
<td>1</td>
</tr>
</tbody>
</table>
</section>
<section id="s2" class="level3">
<h3 class="anchored" data-anchor-id="s2">S2</h3>
<p><strong>Description:</strong> S2 dissimilarity between two sets of labels</p>
<p><strong>Subroutine:</strong> calc_s2</p>
<p><strong>Reference:</strong> <a href="https://doi.org/10.1046/j.0021-8790.2001.00563.x">Lennon et al.&nbsp;(2001) J Animal Ecol.</a></p>
<p><strong>Formula:</strong> <span class="math inline">\(= 1 - \frac{A}{A + min(B, C)}\)</span> where <span class="math inline">\(A\)</span> is the count of labels found in both neighbour sets, <span class="math inline">\(B\)</span> is the count unique to neighbour set 1, and <span class="math inline">\(C\)</span> is the count unique to neighbour set 2. Use the <a href="#label-counts">Label counts</a> calculation to derive these directly.</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Grouping metric?</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>S2</td>
<td>S2 dissimilarity index</td>
<td>cluster metric</td>
<td>2</td>
</tr>
</tbody>
</table>
</section>
<section id="simpson-and-shannon" class="level3">
<h3 class="anchored" data-anchor-id="simpson-and-shannon">Simpson and Shannon</h3>
<p><strong>Description:</strong> Simpson and Shannon diversity metrics using samples from all neighbourhoods.</p>
<p><strong>Subroutine:</strong> calc_simpson_shannon</p>
<p><strong>Formula:</strong> For each index formula, <span class="math inline">\(p_i\)</span> is the number of samples of the i’th label as a proportion of the total number of samples <span class="math inline">\(n\)</span> in the neighbourhoods.</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Minimum number of neighbour sets</em></th>
<th><em>Formula</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>SHANNON_E</td>
<td>Shannon’s evenness (H / HMAX)</td>
<td>1</td>
<td><span class="math inline">\(Evenness = \frac{H}{HMAX}\)</span></td>
</tr>
<tr class="even">
<td>SHANNON_H</td>
<td>Shannon’s H</td>
<td>1</td>
<td><span class="math inline">\(H = - \sum^n_{i=1} (p_i \cdot ln (p_i))\)</span></td>
</tr>
<tr class="odd">
<td>SHANNON_HMAX</td>
<td>maximum possible value of Shannon’s H</td>
<td>1</td>
<td><span class="math inline">\(HMAX = ln(richness)\)</span></td>
</tr>
<tr class="even">
<td>SIMPSON_D</td>
<td>Simpson’s D. A score of zero is more similar.</td>
<td>1</td>
<td><span class="math inline">\(D = 1 - \sum^n_{i=1} p_i^2\)</span></td>
</tr>
</tbody>
</table>
</section>
<section id="sorenson" class="level3">
<h3 class="anchored" data-anchor-id="sorenson">Sorenson</h3>
<p><strong>Description:</strong> Sorenson dissimilarity between two sets of labels. It is the complement of the (unimplemented) Czechanowski index, and numerically the same as Whittaker’s beta.</p>
<p><strong>Subroutine:</strong> calc_sorenson</p>
<p><strong>Formula:</strong> <span class="math inline">\(= 1 - \frac{2A}{2A + B + C}\)</span> where <span class="math inline">\(A\)</span> is the count of labels found in both neighbour sets, <span class="math inline">\(B\)</span> is the count unique to neighbour set 1, and <span class="math inline">\(C\)</span> is the count unique to neighbour set 2. Use the <a href="#label-counts">Label counts</a> calculation to derive these directly.</p>
<p><strong>Indices:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
</colgroup>
<thead>
<tr class="header">
<th><em>Index</em></th>
<th><em>Description</em></th>
<th><em>Grouping metric?</em></th>
<th><em>Minimum number of neighbour sets</em></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>SORENSON</td>
<td>Sorenson index</td>
<td>cluster metric</td>
<td>2</td>
</tr>
</tbody>
</table>


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
      const outerScaffold = trigger.parentElement.cloneNode(true);
      const codeEl = outerScaffold.querySelector('code');
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp('/' + window.location.host + '/');
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
            // target, if specified
            link.setAttribute("target", "_blank");
            if (link.getAttribute("rel") === null) {
              link.setAttribute("rel", "noopener");
            }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./index.html" class="pagination-link" aria-label="Overview">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-title">Overview</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
  </div>
</nav>
</div> <!-- /content -->




</body></html>